---
title: "16s_curacao"
author: "Brad Weiler"
date: "2023-10-04"
output: "/Volumes/projects/weiler/disease/scripts/Curacao_disease_16S.Rmd"
---


#Load in required programs for the pipeline 
```{r, warning=FALSE}
library("devtools"); packageVersion("devtools")
library("phyloseq"); packageVersion("phyloseq")
library("dada2"); packageVersion("dada2")
library("Biostrings"); packageVersion("Biostrings")
library("ggplot2"); packageVersion("ggplot2")
library("ALDEx2");packageVersion("ALDEx2")
library("vegan"); packageVersion("vegan")
library("viridis"); packageVersion("viridis")
library("tidyr"); packageVersion("tidyr")
library("dplyr"); packageVersion("dplyr")
library("scales"); packageVersion("scales")
library("grid"); packageVersion("grid")
library("reshape2"); packageVersion("reshape2")
library("edgeR"); packageVersion("edgeR")
library("plyr"); packageVersion("plyr")
library("DESeq2"); packageVersion("DESeq2")
library("gridExtra"); packageVersion("gridExtra")
library("ampvis2"); packageVersion("ampvis2")
library("phangorn"); packageVersion("phangorn")
library("msa"); packageVersion("msa")
library("ape"); packageVersion("ape")
library("Rcpp"); packageVersion("Rcpp")
library("DECIPHER"); packageVersion("DECIPHER")
library("ggpubr"); packageVersion("ggpubr")
library("knitr"); packageVersion("knitr")
library("RColorBrewer"); packageVersion("RColorBrewer")
library("microbiome"); packageVersion("microbiome")
library("ggjoy"); packageVersion("ggjoy")
library("ggstance"); packageVersion("ggstance")
library("ggridges"); packageVersion("ggridges")
library("ggtree"); packageVersion("ggtree")
library("QsRutils"); packageVersion("QsRutils")
library("phytools"); packageVersion("phytools")
library("nlme"); packageVersion("nlme")
library("tidyverse"); packageVersion("tidyverse")
library("compositions"); packageVersion("compositions")
library("ANCOMBC"); packageVersion("ANCOMBC")
library("DT"); packageVersion("DT")
library("tibble"); packageVersion("tibble")
library("caret"); packageVersion("caret")
library("htmlwidgets"); packageVersion("htmlwidgets")
library("circlize"); packageVersion("circlize")
library("webshot"); packageVersion("webshot")
library("microeco"); packageVersion("microeco")
library("file2meco"); packageVersion("file2meco")
library("rgexf"); packageVersion("rgexf")
library("igraph"); packageVersion("igraph")
library("RColorBrewer")
library("LaCroixColoR")
library("decontam")
library("ggplot2")
library("phyloseq")
library("shadowtext")
library("ggtree")
library("ggtreeExtra")
library("treeio")
library("tidytree")
library("see")
library("gghalves")
library("limorhyde")
#library(curatedMetagenomicData)
library("NetCoMi"); packageVersion("NetCoMi")
library("microeco")
library("htmlwidgets")
library("circlize")
library("chorddiag")
library("webshot")
library("R.matlab")
library("grid")
library("gridBase")
#library(""); packageVersion("")

```

##################################################################################
##                                Tree Building                                 ##
##################################################################################

#Subset the ASVs you want to build your tree with
```{r}
#Make a textfile containing all relevant ASvs in your subsetted phyloseq object from below
write.csv(diel@tax_table, "/Volumes/projects/weiler/curacao/data/16s_diel/diel_taxtable.csv")
```

#Need to use AWK in terminal
```{bash}
#Using that text file we can use AWK to filter those ASVs from the actual fasta file
awk '
FNR==NR{
  arr[$0]
  next
}
/^>/{
  found=""
  match($0,/^>[^.]*/)
  if(substr($0,RSTART+1,RLENGTH-1) in arr){
    found=1
  }
}
found
' FilteredDielAsvs.txt ASVs.fa > FilteredDielAsvs.fa
```

#load in filtered ASVs fasta
```{r}
library(DECIPHER); packageVersion("DECIPHER")
library(phangorn); packageVersion("phangorn")
#load the data
diel_seqs <- readDNAStringSet("/Volumes/projects/weiler/curacao/data/16s_diel/FilteredDielAsvs.fa")
#align the seqs
alignment <- AlignSeqs(DNAStringSet(diel_seqs), anchor=NA)
#phangorn
phang.align <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phang.align)
treeNJ <- NJ(dm)
fit = pml(treeNJ, data=phang.align)
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE, rearrangement="stochastic", control=pml.control(trace = 0))
tree <- fitGTR$tree
write.tree(tree, file = "/Volumes/projects/weiler/curacao/data/16s_diel/diel_tree.nw", append = FALSE, digits = 10, tree.names = FALSE)
```

#RAXML
```{r}
raxmlHPC-PTHREADS-SSE3 -T 4 -m GTRCAT -p 31415 -x 20398 -d -f a -N 100 -n ML_Tree -s "~/analysis/ASVs_aligned.fa"
```

##################################################################################
##                          Build Phyloseq/Filter                               ##
##################################################################################
#Build Phyloseq Object
```{r}
library(phyloseq); packageVersion("phyloseq")
SV <- read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/exported_files/16s_expfiles_diel/ASVs_counts.tsv", row.names = 1, check.names= FALSE, sep = "\t", header = TRUE)
tax <-as.matrix(read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/exported_files/16s_expfiles_diel/ASVs_taxonomy.tsv", row.names = 1, header = TRUE, sep = "\t", fill=TRUE))
map <- read.delim("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/curacao_mapping.txt", sep ="\t", header = TRUE, row.names = 1)
#library(read.table)
#tree <- ape::read.tree(file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/trees/16s_diel_tree.nw")

library(dplyr)
map <- filter(map, project=='Diel') %>% .[,colSums(is.na(map))<nrow(map)]

#Without a tree
ps_diel = phyloseq(otu_table(SV, taxa_are_rows=TRUE), 
               sample_data(map), 
               phyloseq::tax_table(tax))
#With a tree
#ps = phyloseq(otu_table(SV, taxa_are_rows=TRUE), 
#               sample_data(map), 
#               phyloseq::tax_table(tax), phy_tree(tree))


##################################################################################################################################
#PLASTIDS
#library(phyloseq); packageVersion("phyloseq")
SV_pr2 <- read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/exported_files/16s_expfiles_diel_pr2/ASVs_counts.tsv", row.names = 1, check.names= FALSE, sep = "\t", header = TRUE)
tax_pr2 <-as.matrix(read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/exported_files/16s_expfiles_diel_pr2/ASVs_taxonomy.tsv", row.names = 1, header = TRUE, sep = "\t", fill=TRUE))
map <- read.delim("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/curacao_mapping.txt", sep ="\t", header = TRUE, row.names = 1)
#tree <- read.tree(file = "/Volumes/projects/weiler/coralmeta/data/pegasus/single/tree.nw")

#Set to the project you want to explore
library(dplyr)
map <- filter(map, project=='Diel') %>% .[,colSums(is.na(map))<nrow(map)]

ps_plas = phyloseq(otu_table(SV_pr2, taxa_are_rows=TRUE), 
               sample_data(map), 
               phyloseq::tax_table(tax_pr2))


##################################################################################################################################


#NETWORK 16S/18S Compiled with already filtered otu and tax tables
library(phyloseq); packageVersion("phyloseq")
SV_network <- read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/diel/network_otus.csv", row.names = 1, check.names= FALSE, sep = ",", header = TRUE)
tax_network <-as.matrix(read.table("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/diel/network_tax.csv", row.names = 1, header = TRUE, sep = ",", fill=TRUE))
map <- read.delim("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/curacao_mapping.txt", sep ="\t", header = TRUE, row.names = 1)
#library(read.table)
#tree <- ape::read.tree(file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/trees/16s_diel_tree.nw")

library(dplyr)
map <- filter(map, project=='Diel') %>% .[,colSums(is.na(map))<nrow(map)]

#Without a tree
pstr_all_network = phyloseq(otu_table(SV_network, taxa_are_rows=TRUE), 
               sample_data(map), 
               phyloseq::tax_table(tax_network))

#Phyloseq filtering
pstr_nosymb_network <- subset_taxa(pstr_all_network, (Order!="Suessiales"| is.na(Order)))
pstr_symb_network <- subset_taxa(pstr_all_network, (Order=="Suessiales"))


#### DAY OR NIGHT

pstrnight <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T12A' |
                                              pstr_all_network@sam_data$diel_time=='T6A')

pstrday <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T12P' |
                                            pstr_all_network@sam_data$diel_time=='T6P')           


pstrnightnosymb <- subset_samples(pstr_nosymb_network, pstr_nosymb_network@sam_data$diel_time=='T12A' |
                                              pstr_nosymb_network@sam_data$diel_time=='T6A')

pstrdaynosymb <- subset_samples(pstr_nosymb_network, pstr_nosymb_network@sam_data$diel_time=='T12P' |
                                            pstr_nosymb_network@sam_data$diel_time=='T6P')  

########TIMEPOINTS

pstr12a <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T12A')

pstr6a <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T6A')

pstr12p <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T12P')

pstr6p <- subset_samples(pstr_all_network, pstr_all_network@sam_data$diel_time=='T6P')

```


#Filter phyloseq object
```{r}
ps_diel_filtered <- subset_taxa(ps_diel, (Order!="Chloroplast"| is.na(Order)))
ps_diel_filtered <- subset_taxa(ps_diel_filtered, (Family!="Mitochondria"| is.na(Family)))
ps_diel_filtered <- subset_taxa(ps_diel_filtered, (Class!="Embryophyceae"| is.na(Class)))
ps_diel_filtered <- subset_taxa(ps_diel_filtered, (Kingdom!="Eukaryota"| is.na(Kingdom)))
f1<- filterfun_sample(topf(0.99))
wh1 <- genefilter_sample(ps_diel_filtered, f1, A=2)
ps_diel_filtered <- prune_taxa(wh1, ps_diel_filtered)

ps_diel_filtered <- prune_samples(sample_sums(ps_diel_filtered)>=250, ps_diel_filtered)
ps_diel_filtered



#ARCHAEA
#ps_diel_archaea<- subset_taxa(ps_diel_filtered, (Kingdom=="Archaea" | is.na(Kingdom)))
#ps_diel_archaea

#PLASTIDS
ps_plas <- subset_taxa(ps_plas, (Domain=="Eukaryota:plas"))
ps_plas <- subset_taxa(ps_plas, (Division!="Streptophyta:plas"| is.na(Class)))
ps_plas <- subset_taxa(ps_plas, (Order!="Embryophyceae"| is.na(Class)))
#ps_chloro <- subset_taxa(ps, (Order=="Chloroplast"))
ps_plas
#Optional Chloro (3283 ASVs to 748 ASVs)
f1<- filterfun_sample(topf(0.99))
whplas <- genefilter_sample(ps_plas, f1, A=2)
ps_plas_filtered <- prune_taxa(whplas, ps_plas)

#control samples
#ps_diel_filtered <- subset_samples(ps_diel_filtered, ps_diel_filtered@sam_data$Sample_or_Control!='Control Sample')

```


#clean taxonomy table **ps_diel_filtered
```{r}
library(stringr)
tax.clean <- ps_diel_filtered@tax_table

tax.clean <- data.frame(row.names = row.names(ps_diel_filtered@tax_table),
Kingdom = str_replace(ps_diel_filtered@tax_table[,1], "D_0__",""),
Phylum = str_replace(ps_diel_filtered@tax_table[,2], "D_1__",""),
Class = str_replace(ps_diel_filtered@tax_table[,3], "D_2__",""),
Order = str_replace(ps_diel_filtered@tax_table[,4], "D_3__",""),
Family = str_replace(ps_diel_filtered@tax_table[,5], "D_4__",""),
Genus = str_replace(ps_diel_filtered@tax_table[,6], "D_5__",""),
Species = str_replace(ps_diel_filtered@tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == ""){kingdom <- paste(tax.clean[i,1],"_X", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == ""){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == ""){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == ""){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == ""){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == ""){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


tax_table(ps_diel_filtered) <- as.matrix(tax.clean)
```

#clean taxonomy table **ps_plas** 
```{r}
library(stringr)
tax.clean <- ps_plas_filtered@tax_table

tax.clean <- data.frame(row.names = row.names(ps_plas_filtered@tax_table),
Domain = str_replace(ps_plas_filtered@tax_table[,1], "D_0__",""),
Supergroup = str_replace(ps_plas_filtered@tax_table[,2], "D_1__",""),
Division = str_replace(ps_plas_filtered@tax_table[,3], "D_2__",""),
Subdivision = str_replace(ps_plas_filtered@tax_table[,4], "D_3__",""),
Class = str_replace(ps_plas_filtered@tax_table[,5], "D_4__",""),
Order = str_replace(ps_plas_filtered@tax_table[,6], "D_5__",""),
Family = str_replace(ps_plas_filtered@tax_table[,7], "D_6__",""),
Genus = str_replace(ps_plas_filtered@tax_table[,8], "D_7__",""),
Species = str_replace(ps_plas_filtered@tax_table[,9], "D_8__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:9){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == ""){domain <- paste(tax.clean[i,1],"_X", sep = "")
  tax.clean[i, 2] <- domain
  tax.clean[i, 3] <- paste(domain,"X", sep = "")
  tax.clean[i, 4] <- paste(domain,"XX", sep = "")
  tax.clean[i, 5] <- paste(domain,"XXX", sep = "")
  tax.clean[i, 6] <- paste(domain,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(domain,"XXXXX", sep = "")
  tax.clean[i, 8] <- paste(domain,"XXXXXX", sep = "")
  tax.clean[i, 9] <- paste(domain,"XXXXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == ""){supergroup <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- supergroup
  tax.clean[i, 4] <- paste(supergroup,"X", sep = "")
  tax.clean[i, 5] <- paste(supergroup,"XX", sep = "")
  tax.clean[i, 6] <- paste(supergroup,"XXX", sep = "")
  tax.clean[i, 7] <- paste(supergroup,"XXXX.", sep = "")
  tax.clean[i, 8] <- paste(supergroup,"XXXXX", sep = "")
  tax.clean[i, 9] <- paste(supergroup,"XXXXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == ""){division <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- division
  tax.clean[i, 5] <- paste(division,"X", sep = "")
  tax.clean[i, 6] <- paste(division,"XX", sep = "")
  tax.clean[i, 7] <- paste(division,"XXX", sep = "")
  tax.clean[i, 8] <- paste(division,"XXXX", sep = "")
  tax.clean[i, 9] <- paste(division,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == ""){subdivision <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- subdivision
  tax.clean[i, 6] <- paste(subdivision,"X", sep = "")
  tax.clean[i, 7] <- paste(subdivision,"XX", sep = "")
  tax.clean[i, 8] <- paste(subdivision,"XXX", sep = "")
  tax.clean[i, 9] <- paste(subdivision,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,6] == ""){class <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- class
  tax.clean[i, 7] <- paste(class,"X", sep = "")
  tax.clean[i, 8] <- paste(class,"XX", sep = "")
  tax.clean[i, 9] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,7] == ""){order <- paste(tax.clean[i,6], "_X", sep = "")
  tax.clean[i, 7] <- order
  tax.clean[i, 8] <- paste(order,"X", sep = "")
  tax.clean[i, 9] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,8] == ""){family <- paste(tax.clean[i,7], "_X", sep = "")
  tax.clean[i, 8] <- family
  tax.clean[i, 9] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,9] == ""){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


tax_table(ps_plas_filtered) <- as.matrix(tax.clean)
```

#export tax tables to infer phylogeny in plots
```{r}
write.csv(ps_diel_filtered@tax_table, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_filtered_taxtable.csv")

```

#Control Exploration
```{r}
ps_control<-subset_samples(ps, ps@sam_data$Sample_or_Control == "Control Sample")


bub_control <- merge_samples(ps_control, ps_control@sam_data$library_name) ##Variable
type_taxa_control <- bub_control %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_control <- data.table(type_taxa_control)
# convert Phylum to a character vector from a factor because R
dat_control$Genus <- as.character(dat_control$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_control[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_control[(median <= 0.001), Genus := "Remainder"]

bp_control <- ggplot(dat_control[Abundance > 0], aes(x = Sample, y = Genus)) + 
  geom_point(aes(size = Abundance, fill=Phylum), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Bacterial Genus", size = "Relative Abundance %", fill="Phylum") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_control

pdf("/Volumes/projects/weiler/curacao/analysis/bp_control.pdf", width = 13, height = 12)
bp_control
dev.off()

write.table(type_taxa_control, file = "/Volumes/projects/weiler/curacao/data/type_taxa_control.tsv")





bub_control_sd <- ps_control %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, VAR1)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bub_control_sd <- merge_samples(ps_control, "VAR1") 
type_taxa1 <- bub1 %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa1 <- type_taxa1 %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat1 <- data.table(type_taxa1)

dat1$SD <- sd1$sd
# convert Phylum to a character vector from a factor because R
dat1$TaxRANK <- as.character(dat1$TaxRANK)
# group dataframe by Phylum, calculate median rel. abundance
dat1[, median := median(Abundance, na.rm = FALSE), 
    by = "TaxRANK"]
# Change name to remainder of Phylum less than 2%
dat1[(median <= 0.01), TaxRANK := "Below Threshold"]
dat1[(TaxRANK == "Below Threshold"), TaxRANK_Higher := "Other"]

my_title1 <- expression(paste("Microbial Orders within ", italic("P. clavata")))

Fig1A <- ggplot(dat1[Abundance > 0], aes(x = factor(Sample, levels=ordered), y = factor(Order, levels=bac_orders))) + geom_point(aes(size = Abundance, fill=TaxRANK_Higher, color=TaxRANK_Higher), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Var1", y = "Prokaryotic TaxRANK", size = "Relative Abundance %", fill="Prokaryotic TaxRANK_Higher") + theme_bw() +  
  scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=my_title1, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Phylum, color=TaxRANK_Higher), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = TaxRANK_Higher, color=TaxRANK_Higher), alpha = 1, shape=21) + guides(color = FALSE, fill = guide_legend(override.aes = list(size = 3)))
Fig1A
```

#Chloro Exploration
```{r}
colnames(ps_chloro@tax_table) <- c("Domain",	"Supergroup",	"Division",	"Subdivision",	"Class",	"Order",	"Family",	"Genus",	"Species")

ps_chloro_diel <- subset_samples(ps_chloro, ps_chloro@sam_data$project=='Diel')
ps_chloro_diel <- subset_samples(ps_chloro_diel, ps_chloro_diel@sam_data$host!='Water')


bub_chloro_diel <- merge_samples(ps_chloro_diel, ps_chloro_diel@sam_data$diel_time) ##Variable
type_taxa_chloro_diel <- bub_chloro_diel %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_chloro_diel <- data.table(type_taxa_chloro_diel)
# convert Phylum to a character vector from a factor because R
dat_chloro_diel$Genus <- as.character(dat_chloro_diel$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_chloro_diel[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_chloro_diel[(median <= 0.001), Genus := "Remainder"]

bp_chloro_diel <- ggplot(dat_chloro_diel[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) + 
  geom_point(aes(size = Abundance, fill=Division), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Plastid Diversity", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_chloro_diel

pdf("/Volumes/projects/weiler/curacao/analysis/bp_chloro_diel.pdf", width = 13, height = 12)
bp_chloro_diel
dev.off()


#PSTR
ps_chloro_diel_pstr <- subset_samples(ps_chloro_diel, ps_chloro_diel@sam_data$host=='Pseudodiploria strigosa')
bub_chloro_diel_pstr <- merge_samples(ps_chloro_diel_pstr, ps_chloro_diel_pstr@sam_data$diel_time) ##Variable
type_taxa_chloro_diel_pstr <- bub_chloro_diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_chloro_diel_pstr <- data.table(type_taxa_chloro_diel_pstr)
# convert Phylum to a character vector from a factor because R
dat_chloro_diel_pstr$Genus <- as.character(dat_chloro_diel_pstr$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_chloro_diel_pstr[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_chloro_diel_pstr[(median <= 0.001), Genus := "Remainder"]

bp_chloro_diel_pstr <- ggplot(dat_chloro_diel_pstr[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) + 
  geom_point(aes(size = Abundance, fill=Division), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Plastid Diversity", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_chloro_diel_pstr

pdf("/Volumes/projects/weiler/curacao/analysis/bp_chloro_diel_pstr.pdf", width = 13, height = 12)
bp_chloro_diel_pstr
dev.off()


#OANN
ps_chloro_diel_oann <- subset_samples(ps_chloro_diel, ps_chloro_diel@sam_data$host=='Orbicella annularis')
bub_chloro_diel_oann <- merge_samples(ps_chloro_diel_oann, ps_chloro_diel_oann@sam_data$diel_time) ##Variable
type_taxa_chloro_diel_oann <- bub_chloro_diel_oann %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_chloro_diel_oann <- data.table(type_taxa_chloro_diel_oann)
# convert Phylum to a character vector from a factor because R
dat_chloro_diel_oann$Genus <- as.character(dat_chloro_diel_oann$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_chloro_diel_oann[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_chloro_diel_oann[(median <= 0.001), Genus := "Remainder"]

bp_chloro_diel_oann <- ggplot(dat_chloro_diel_oann[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) + 
  geom_point(aes(size = Abundance, fill=Division), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Plastid Diversity", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_chloro_diel_oann

pdf("/Volumes/projects/weiler/curacao/analysis/bp_chloro_diel_oann.pdf", width = 13, height = 12)
bp_chloro_diel_oann
dev.off()


#DLAB
ps_chloro_diel_dlab <- subset_samples(ps_chloro_diel, ps_chloro_diel@sam_data$host=='Diploria labyrinthiformis')
bub_chloro_diel_dlab <- merge_samples(ps_chloro_diel_dlab, ps_chloro_diel_dlab@sam_data$diel_time) ##Variable
type_taxa_chloro_diel_dlab <- bub_chloro_diel_dlab %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_chloro_diel_dlab <- data.table(type_taxa_chloro_diel_dlab)
# convert Phylum to a character vector from a factor because R
dat_chloro_diel_dlab$Genus <- as.character(dat_chloro_diel_dlab$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_chloro_diel_dlab[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_chloro_diel_dlab[(median <= 0.001), Genus := "Remainder"]

bp_chloro_diel_dlab <- ggplot(dat_chloro_diel_dlab[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) + 
  geom_point(aes(size = Abundance, fill=Division), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Plastid Diversity", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_chloro_diel_dlab

pdf("/Volumes/projects/weiler/curacao/analysis/bp_chloro_diel_dlab.pdf", width = 13, height = 12)
bp_chloro_diel_dlab
dev.off()

#Water
ps_chloro_diel_water <- subset_samples(ps_chloro, ps_chloro@sam_data$project=='Diel')
ps_chloro_diel_water <- subset_samples(ps_chloro_diel_water, ps_chloro_diel_water@sam_data$host=='Water')
bub_chloro_diel_water <- merge_samples(ps_chloro_diel_water, ps_chloro_diel_water@sam_data$diel_time) ##Variable
type_taxa_chloro_diel_water <- bub_chloro_diel_water %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% 
  arrange(Genus) 

dat_chloro_diel_water <- data.table(type_taxa_chloro_diel_water)
# convert Phylum to a character vector from a factor because R
dat_chloro_diel_water$Genus <- as.character(dat_chloro_diel_water$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_chloro_diel_water[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_chloro_diel_water[(median <= 0.001), Genus := "Remainder"]

bp_chloro_diel_water <- ggplot(dat_chloro_diel_water[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) +
  geom_point(aes(size = Abundance, fill=Division), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Sample", y = "Plastid Diversity", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  scale_fill_manual(values = palette, guide = guide_legend(override.aes = list(size = 3, alpha = 1))) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

bp_chloro_diel_water

pdf("/Volumes/projects/weiler/curacao/analysis/bp_chloro_diel_water.pdf", width = 13, height = 12)
bp_chloro_diel_water
dev.off()

```

#Decontamination of samples *Custom Workflow*
```{r}
library("decontam")
sample_data(ps_filtered)$is.neg <- ps_filtered@sam_data[["Sample_or_Control"]] == "Control Sample"
contamdf.prev <- isContaminant(ps_filtered, method="prevalence", neg="is.neg")
table(contamdf.prev$contaminant)
head(which(contamdf.prev$contaminant))

write.csv(contamdf.prev, "/Volumes/projects/weiler/curacao/data/contaminant_asvs.csv")

#contamdf.prev05 <- isContaminant(ps_filtered, method="prevalence", neg="is.neg", threshold=0.5)
#table(contamdf.prev05$contaminant)

# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(ps_filtered, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "Control Sample", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "True Sample", ps.pa)
# Make data.frame of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")

#identifying the bad ASVs to be removed
badTaxa <- c("ASV_4", "ASV_5", "ASV_6", "ASV_7", "ASV_10", "ASV_14", "ASV_15", "ASV_16", "ASV_17", "ASV_23", "ASV_28", "ASV_29", "ASV_33", "ASV_42", "ASV_46", "ASV_60", "ASV_63", "ASV_84", "ASV_130", "ASV_141", "ASV_144", "ASV_148", "ASV_149", "ASV_150", "ASV_177", "ASV_188", "ASV_191", "ASV_262", "ASV_292", "ASV_301", "ASV_312", "ASV_370", "ASV_387", "ASV_438", "ASV_713", "ASV_760", "ASV_807", "ASV_926", "ASV_998", "ASV_1010", "ASV_1227", "ASV_1357", "ASV_1462", "ASV_1508", "ASV_1510", "ASV_1644", "ASV_1699", "ASV_1789", "ASV_1972", "ASV_1973", "ASV_1998", "ASV_2124", "ASV_2163", "ASV_2169", "ASV_2188", "ASV_2252", "ASV_2373", "ASV_2413", "ASV_2414", "ASV_2493", "ASV_2510", "ASV_2571", "ASV_2576", "ASV_2608", "ASV_2630", "ASV_2664", "ASV_2710", "ASV_2999", "ASV_3029", "ASV_3197", "ASV_3218", "ASV_3413", "ASV_3440", "ASV_3627", "ASV_3634", "ASV_3797", "ASV_3913", "ASV_3943", "ASV_3992", "ASV_4005", "ASV_4015", "ASV_4018", "ASV_4022", "ASV_4029", "ASV_4077", "ASV_4640", "ASV_4780", "ASV_4844", "ASV_4879", "ASV_4997", "ASV_5023", "ASV_5066", "ASV_5086", "ASV_5162", "ASV_5418", "ASV_5467", "ASV_5643", "ASV_5824", "ASV_5957", "ASV_6087", "ASV_6139", "ASV_6168", "ASV_6244", "ASV_6329", "ASV_6503", "ASV_6549", "ASV_6557", "ASV_6689", "ASV_6698", "ASV_6758", "ASV_6812", "ASV_6920", "ASV_7056", "ASV_7336", "ASV_7473", "ASV_7516", "ASV_7607", "ASV_7791", "ASV_7880", "ASV_7898", "ASV_7904", "ASV_7957", "ASV_7984", "ASV_8512", "ASV_8664", "ASV_8803", "ASV_9363", "ASV_9672", "ASV_10502", "ASV_10840", "ASV_11255", "ASV_11504", "ASV_12041", "ASV_13006", "ASV_13829", "ASV_13924", "ASV_13926", "ASV_15490", "ASV_15974", "ASV_18043", "ASV_18910", "ASV_19156", "ASV_20530")

#getting the taxa names from the phyloseq
allTaxa <- taxa_names(ps_filtered)
allTaxa_chloro <- taxa_names(ps_chloro)
allTaxa_mito <- taxa_names(ps_mito)

#removing the good taxa from the taxa names
allTaxa <- allTaxa[!(allTaxa %in% badTaxa)]
allTaxa_chloro <- allTaxa_chloro[!(allTaxa_chloro %in% badTaxa)]
allTaxa_mito <- allTaxa_mito[!(allTaxa_mito %in% badTaxa)]
# new phyloseq object with excluded ASVs
ps_filtered <- prune_taxa(allTaxa, ps_filtered)
ps_chloro <- prune_taxa(allTaxa_chloro, ps_chloro)
ps_mito <- prune_taxa(allTaxa_mito, ps_mito)

# add in new phyloseq object is renamed and not rewritten
ps_filtered

ps_filtered <- subset_samples(ps_filtered, ps_filtered@sam_data$Sample_or_Control=="True Sample")

#filtered ASV taxonomy tables
write.table(ps_filtered@tax_table, file = "/Volumes/projects/weiler/curacao/data/ps_filtered_taxtable.tsv")
write.table(ps_mito@tax_table, file = "/Volumes/projects/weiler/curacao/data/ps_mito_taxtable.tsv")
write.table(ps_chloro@tax_table, file = "/Volumes/projects/weiler/curacao/data/ps_chloro_taxtable.tsv")
```

#Using script to pull ASVs from a fasta file based on a text file
```{bash}
awk '
FNR==NR{
  arr[$0]
  next
}
/^>/{
  found=""
  match($0,/^>[^.]*/)
  if(substr($0,RSTART+1,RLENGTH-1) in arr){
    found=1
  }
}
found
' FilteredChloroAsvs.txt ASVs.fa > FilteredChloroAsvs.fa


awk '
FNR==NR{
  arr[$0]
  next
}
/^>/{
  found=""
  match($0,/^>[^.]*/)
  if(substr($0,RSTART+1,RLENGTH-1) in arr){
    found=1
  }
}
found
' FilteredMitosAsvs.txt ASVs.fa > FilteredMitosAsvs.fa

```

#Assigning taxonomy to subsetted fasta files
```{r}
seqtabchloro<- read_file("/Volumes/projects/weiler/curacao/data/FilteredMitosAsvs.fa")

ref_fasta <- "/Volumes/projects/weiler/curacao/data/pr2_version_5.0.0_SSU_dada2.fasta.gz"
taxa <- assignTaxonomy(seqtabchloro, refFasta=ref_fasta, multithread=2, taxLevels = c("Kingdom","Supergroup","Division","Class","Order","Family","Genus","Species"))
colnames(taxa) <- c("Kingdom", "Supergroup","Division", "Class", "Order", "Family", "Genus", "Species")

#make blast database
makeblastdb -in /Volumes/projects/weiler/curacao/data/pr2_version_5.0.0_SSU_dada2.fasta -dbtype nucl -out pr2

pr2 <- "/Volumes/projects/weiler/curacao/data/pr2_version_5.0.0_SSU_dada2.fasta.gz"
blastn -task megablast -num_threads 4 -outfmt "6 qseqid qlen length bitscore qcovs evalue pident sseqid" -db /Volumes/projects/weiler/curacao/data/pr2_version_5.0.0_SSU_dada2.fasta.gz -max_target_seqs 1 -query FilteredMitosAsvs.fa -out ASVs_chloro.out

#BLAST NAs Present
blastn -task megablast -num_threads 4 -outfmt "6 qseqid qlen length bitscore qcovs evalue pident sseqid" -db pr2 -max_target_seqs 1 -query ASVs_NAs.fasta -out ASVs_NAs.out
```




#Subset phyloseq objects **Ensure that no control samples are included in dataset**
```{r}
#Sorting out character/factor data type
#ps_filtered@sam_data$project <- as.factor(ps_filtered@sam_data$project)
#ps_filtered@sam_data$host <- as.factor(ps_filtered@sam_data$host)

#DIEL
diel <- ps_diel_filtered

diel_w <- subset_samples(diel, diel@sam_data$host=='Water')

diel_pstrw <- subset_samples(diel, diel@sam_data$host=='Pseudodiploria strigosa' |
                                      diel@sam_data$host=='Water')
#dielsymb_pstrw <- subset_samples(dielsymb, diel@sam_data$host=='Pseudodiploria strigosa' |
#                                      diel@sam_data$host=='Water')

#diel_oannw <- subset_samples(diel, diel@sam_data$host=='Orbicella annularis' |
#                                      diel@sam_data$host=='Water')
#diel_dlabw <- subset_samples(diel, diel@sam_data$host=='Diploria labyrinthiformis' |
#                                      diel@sam_data$host=='Water')

diel_pstr <- subset_samples(diel, diel@sam_data$host=='Pseudodiploria strigosa')
diel_pstr <- prune_taxa(taxa_sums(diel_pstr) > 0, diel_pstr)

#diel_oann <- subset_samples(diel, diel@sam_data$host=='Orbicella annularis')
#diel_dlab <- subset_samples(diel, diel@sam_data$host=='Diploria labyrinthiformis')
#diel_water <- subset_samples(diel, diel@sam_data$host=='Water')


#PLASTIDS
diel_pstrw_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Pseudodiploria strigosa' |
                                      ps_plas_filtered@sam_data$host=='Water')
diel_oannw_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Orbicella annularis' |
                                      ps_plas_filtered@sam_data$host=='Water')
diel_dlabw_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Diploria labyrinthiformis' |
                                      diel@sam_data$host=='Water')

diel_pstr_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Pseudodiploria strigosa')
diel_oann_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Orbicella annularis')
diel_dlab_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Diploria labyrinthiformis')
diel_water_plas <- subset_samples(ps_plas_filtered, ps_plas_filtered@sam_data$host=='Water')




##################################################################################
##################################################################################
###                                  Ordering                                  ###
##################################################################################
##################################################################################
library(dplyr)
#Bacterial Taxonomy
bact_class <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_bact_class.txt",header=F)
bactclass <- bact_class %>% pull(V1)
bact_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_bact_order.txt",header=F)
bactorder <- bact_order %>% pull(V1)
bact_family <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_bact_family.txt",header=F)
bactfamily <- bact_family %>% pull(V1)
bact_genus <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_bact_genus.txt",header=F)
bactgenus <- bact_genus %>% pull(V1)
bact_species <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_bact_species.txt",header=F)
bactspecies <- bact_genus %>% pull(V1)

#Eukaryotic Taxonomy
euk_supergroup <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_supergroup.txt",header=F)
euksupergroup <- euk_supergroup %>% pull(V1)
euk_division <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_division.txt",header=F)
eukdivision <- euk_division %>% pull(V1)
euk_subdivision <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_subdivision.txt",header=F)
euksubdivision <- euk_subdivision %>% pull(V1)
euk_class <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_class.txt",header=F)
eukclass <- euk_class %>% pull(V1)
euk_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_order.txt",header=F)
eukorder <- euk_order %>% pull(V1)
euk_family <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_family.txt",header=F)
eukfamily <- euk_family %>% pull(V1)
euk_genus <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_genus.txt",header=F)
eukgenus <- euk_genus %>% pull(V1)
euk_species <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_euk_species.txt",header=F)
eukspecies <- euk_species %>% pull(V1)


#Plastid Taxonomy
plast_species <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_species.txt",header=F)
plast_species <- plast_species %>% pull(V1)
plast_genus <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_genus.txt",header=F)
plast_genus <- plast_genus %>% pull(V1)
plast_family <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_family.txt",header=F)
plast_family <- plast_family %>% pull(V1)
plast_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_order.txt",header=F)
plast_order <- plast_order %>% pull(V1)
plast_class <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_class.txt",header=F)
plast_class <- plast_class %>% pull(V1)
plast_subdivision <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_plast_subdivision.txt",header=F)
plast_subdivision <- plast_subdivision %>% pull(V1)

#By Structure
time_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/time_order.txt",header=F)
timeorder <- time_order %>% pull(V1)
daytime_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/daytime_order.txt",header=F)
daytimeorder <- daytime_order %>% pull(V1)
diel_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/diel_order.txt",header=F)
dielorder <- diel_order %>% pull(V1)
dielpstr_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dielpstr_order.txt",header=F)
dielpstrorder <- dielpstr_order %>% pull(V1)
#dieloann_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dieloann_order.txt",header=F)
#dieloannorder <- dieloann_order %>% pull(V1)
#dieldlab_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dieldlab_order.txt",header=F)
#dieldlaborder <- dieldlab_order %>% pull(V1)
dielpstr_day_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dielpstr_day_order.txt",header=F)
dielpstrdayorder <- dielpstr_day_order %>% pull(V1)
#dieloann_day_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dieloann_day_order.txt",header=F)
#dieloanndayorder <- dieloann_day_order %>% pull(V1)
#dieldlab_day_order <- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/ordering/dieldlab_day_order.txt",header=F)
#dieldlabdayorder <- dieldlab_day_order %>% pull(V1)
```



#SUBSET PHYLOSEQ FOR SUBPHYLO OTU EXPLORATION
```{r}

#OTU Tables extraction
OTU = as(otu_table(ps_diel), "matrix") 
#OTU = t(OTU)
OTU = as.data.frame(OTU)
write.table(OTU, file = "/Volumes/projects/weiler/curacao/data/16s_diel/diel_otutable.tsv")
#TAX Table extraction
TAX = as(phyloseq::tax_table(ps_diel), "matrix")
TAX = as.data.frame(TAX)
write.table(TAX, file = "/Volumes/projects/weiler/curacao/data/16s_diel/diel_taxtable.tsv")
#OTU Table w/ Tax Assignment extraction
OTU <- tibble::rownames_to_column(OTU, "ASV")
TAX <- tibble::rownames_to_column(TAX, "ASV")
OTUTAX<- merge(OTU, TAX, by= "ASV")
write.table(OTUTAX, file = "/Volumes/projects/weiler/curacao/data/16s_diel/diel_otutaxtable.tsv")



#PSTR
#OTU Tables extraction
OTU = as(otu_table(diel_pstr), "matrix") 
OTU = t(OTU)
OTU = as.data.frame(OTU)
write.table(OTU, file = "/Volumes/projects/weiler/curacao/data/16s_diel/pstr/diel_pstr_otutable.tsv")
#TAX Table extraction
TAX = as(tax_table(diel_pstr), "matrix")
TAX = as.data.frame(TAX)
write.table(TAX, file = "/Volumes/projects/weiler/curacao/data/16s_diel/pstr/diel_pstr_taxtable.tsv")
#OTU Table w/ Tax Assignment extraction
OTU <- tibble::rownames_to_column(OTU, "ASV")
TAX <- tibble::rownames_to_column(TAX, "ASV")
OTUTAX<- merge(OTU, TAX, by= "ASV")
write.table(OTUTAX, file = "/Volumes/projects/weiler/curacao/data/16s_diel/pstr/diel_pstr_otutaxtable.tsv")
#write.table(OTUTAX, file = "~/Desktop/diel_pstr_otutaxtable.tsv")




```

##################################################################################
##                            Colour Palettes                                   ##
##################################################################################

# Colour Palettes
```{r}
library(randomcoloR)
n <- 64
palette <- distinctColorPalette(n)
c28 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "darkorange4", "brown", "firebrick3", "darkslategray", "gray0"
)


day_palette <- c("midnightblue", "gold")
diel_palette <- c("midnightblue", "slategray", "gold", "orange")
diel_palette_ordered <- c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange")

safe_colorblind_palette <- c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", 
                             "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")
                             

palette_all <- c("honeydew", "honeydew1", "honeydew2", "honeydew3", "honeydew4", "hotpink", "hotpink1", "hotpink2", "hotpink3", "hotpink4", "indianred", "indianred1", "indianred2", "indianred3", "indianred4", "ivory", "ivory1", "ivory2", "ivory3", "ivory4", "khaki", "khaki1", "khaki2", "khaki3", "khaki4", "lavender", "lavenderblush", "lavenderblush1", "lavenderblush2", "lavenderblush3", "lavenderblush4", "lawngreen", "lemonchiffon", "lemonchiffon1", "lemonchiffon2", "lemonchiffon3", "lemonchiffon4", "lightblue", "lightblue1", "lightblue2", "lightblue3", "lightblue4", "lightcoral", "lightcyan", "lightcyan1", "lightcyan2", "lightcyan3", "lightcyan4", "lightgoldenrod", "lightgoldenrod1", "lightgoldenrod2", "lightgoldenrod3", "lightgoldenrod4", "lightgoldenrodyellow", "lightgray", "lightgreen", "lightgrey", "lightpink", "lightpink1", "lightpink2", "lightpink3", "lightpink4", "lightsalmon", "lightsalmon1", "lightsalmon2", "lightsalmon3", "lightsalmon4", "lightseagreen", "lightskyblue", "lightskyblue1", "lightskyblue2", "lightskyblue3", "lightskyblue4", "lightslateblue", "lightslategray", "lightslategrey", "lightsteelblue", "lightsteelblue1", "lightsteelblue2", "lightsteelblue3", "lightsteelblue4", "lightyellow", "lightyellow1", "lightyellow2", "lightyellow3", "lightyellow4", "limegreen", "linen", "magenta", "magenta1", "magenta2", "magenta3", "magenta4", "maroon", "maroon1", "maroon2", "maroon3", "maroon4", "mediumaquamarine", "mediumblue", "mediumorchid", "mediumorchid1", "mediumorchid2", "mediumorchid3", "mediumorchid4", "mediumpurple", "mediumpurple1", "mediumpurple2", "mediumpurple3", "mediumpurple4", "mediumseagreen", "mediumslateblue", "mediumspringgreen", "mediumturquoise", "mediumvioletred", "midnightblue", "mintcream", "mistyrose", "mistyrose1", "mistyrose2", "mistyrose3", "mistyrose4", "moccasin", "navajowhite", "navajowhite1", "navajowhite2", "navajowhite3", "navajowhite4", "navy", "navyblue", "oldlace", "olivedrab", "olivedrab1", "olivedrab2", "olivedrab3", "olivedrab4", "orange", "orange1", "orange2", "orange3", "orange4", "orangered", "orangered1", "orangered2", "orangered3", "orangered4", "orchid", "orchid1", "orchid2", "orchid3", "orchid4", "palegoldenrod", "palegreen", "palegreen1", "palegreen2", "palegreen3", "palegreen4", "paleturquoise", "paleturquoise1", "paleturquoise2", "paleturquoise3", "paleturquoise4", "palevioletred", "palevioletred1", "palevioletred2", "palevioletred3", "palevioletred4", "papayawhip", "peachpuff", "peachpuff1", "peachpuff2", "peachpuff3", "peachpuff4", "peru", "pink", "pink1", "pink2", "pink3", "pink4", "plum", "plum1", "plum2", "plum3", "plum4", "powderblue", "purple", "purple1", "purple2", "purple3", "purple4", "red", "red1", "red2", "red3", "red4", "rosybrown", "rosybrown1", "rosybrown2", "rosybrown3", "rosybrown4", "royalblue", "royalblue1", "royalblue2", "royalblue3", "royalblue4", "saddlebrown", "salmon", "salmon1", "salmon2", "salmon3", "salmon4", "sandybrown", "seagreen", "seagreen1", "seagreen2", "seagreen3", "seagreen4", "seashell", "seashell1", "seashell2", "seashell3", "seashell4", "sienna", "sienna1", "sienna2", "sienna3", "sienna4", "skyblue", "skyblue1", "skyblue2", "skyblue3", "skyblue4", "slateblue", "slateblue1", "slateblue2", "slateblue3", "slateblue4", "slategray", "slategray1", "slategray2", "slategray3", "slategray4", "slategrey", "snow", "snow1", "snow2", "snow3", "snow4", "springgreen", "springgreen1", "springgreen2", "springgreen3", "springgreen4", "steelblue", "steelblue1", "steelblue2", "steelblue3", "steelblue4", "tan", "tan1", "tan2", "tan3", "tan4", "thistle", "thistle1", "thistle2", "thistle3", "thistle4", "tomato", "tomato1", "tomato2", "tomato3", "tomato4", "turquoise", "turquoise1", "turquoise2", "turquoise3", "turquoise4", "violet", "violetred", "violetred1", "violetred2", "violetred3", "violetred4", "wheat", "wheat1", "wheat2", "wheat3", "wheat4", "whitesmoke", "yellow", "yellow1", "yellow2", "yellow3", "yellow4", "yellowgreen")

eukdivision_palette <- c("Other"="slategrey","Stramenopiles"="skyblue","Alveolata"="sienna","Chlorophyta"="seagreen")
eukclass_palette <- c('Other'="slategrey", 'Bacillariophyceae'="#CC6677", 'Syndiniales'="#661100",'Dinophyceae'="#88CCEE", 'Oligohymenophorea'="#999933", 'Coccidiomorphea'="#117733", 'Ulvophyceae'="#DDCC77")
euksymbfamily_palette <- c("Other"="slategrey","Symbiodiniaceae"="#DDCC77")
euksymbgenus_palette <- c("Other"="slategrey","Durusdinium"="#DDCC77","Cladocopium"="#999933","Breviolum"="#661100","Symbiodinium"="#6699CC")
plassubdiv_palette <- c("Other"="slategrey","Gyrista:plas"="#DDCC77","Chrompodellids:plas"="#999933","Haptophyta_X:plas"="#661100","Chlorophyta_X:plas"="#6699CC")

```

# Colour Palettes
```{r}
install.packages("devtools") 
devtools::install_github("johannesbjork/LaCroixColoR")

library(LaCroixColoR)

#Discrete Palettes
lacroix_palette("Pamplemousse", type = "discrete")
#lacroix_palette("PassionFruit", type = "discrete")
#lacroix_palette("PeachPear", type = "discrete")

#Continuous
#lacroix_palette("Pamplemousse", n = 50, type = "continuous")
#lacroix_palette("PassionFruit", n = 50, type = "continuous")
#lacroix_palette("PassionFruit", n = 25, type = "continuous")
#lacroix_palette("PassionFruit", n = 10, type = "continuous")
#lacroix_palette("PeachPear", n = 50, type = "continuous")

PassionFruit = rbind(c("#C70E7B", "#FC6882","#A6E000","#1BB6AF","#6C6C9D","#172869"),c(1,3,2,4,5,6)),
  Mango = rbind(c("#FF5300","#9ED80B","#43B629","#1BB6AF","#8F92A1","#172869"), c(1,2,6,3,5,4)),
  Pure = rbind(c("#AFDFEF","#54BCD1","#1BB6AF","#0099D5","#007BC3","#172869"), c(1,3,5,2,4,6)),
  Lime = rbind(c("#2CB11B","#95C65C","#BDDE9B","#1BB6AF","#0076C0","#172869"),c(1,3,5,2,4,6)),
  Lemon = rbind(c("#F7AA14","#F5D000","#F7E690","#1BB6AF","#088BBE","#172869"),c(1,3,5,2,4,6)),
  Orange = rbind(c("#EF7C12","#FCA315","#F4B95A","#1BB6AF","#088BBE","#172869"),c(1,3,4,2,5,6)),
  Berry = rbind(c("#B25D91","#CB87B4","#EFC7E6","#1BB6AF","#088BBE","#172869"),c(1,3,4,2,5,6)),
  CranRaspberry = rbind(c("#D9565C","#F28A8A","#EDA9AB","#1BB6AF","#088BBE","#172869"),c(1,3,4,2,5,6)),
  Pamplemousse = rbind(c("#EA7580","#F6A1A5","#F8CD9C","#1BB6AF","#088BBE","#172869"),c(1,3,2,5,4,6)),
  PeachPear = rbind(c("#FF3200","#E9A17C","#E9E4A6","#1BB6AF","#0076BB","#172869"),c(1,3,5,2,4,6)),
  Coconut = rbind(c("#881C00","#AF6125","#F4E3C7","#1BB6AF","#0076BB","#172869"),c(1,3,5,2,4,6)),
  Apricot = rbind(c("#D72000","#EE6100","#FFAD0A","#1BB6AF","#9093A2","#132157"),c(1,3,4,2,5,6)),
  Tangerine = rbind(c("#EF562A","#EC921D","#F7B449","#FFED00","#1BB6AF","#9093A2","#132157"), c(1,4,5,3,6,2,7)),
  KeyLime = rbind(c("#D84D16","#FFF800","#8FDA04","#009F3F","#132157"),c(3,1,2,4,5)),
  PommeBaya = rbind(c("#C23A4B","#FBBB48","#EFEF46","#31D64D","#132157"),c(1,4,2,5,3)),
  CeriseLimon =rbind( c("#EE4244","#F8D961","#B6D944","#638E6E","#3C5541", "#132157"),c(1,3,5,2,6,4)),
  PinaFraise = rbind(c("#F44B4B","#F19743","#F1F1A8","#92D84F","#7473A6", "#132157"),c(1,4,2,5,3,6)),
  KiwiSandia = rbind(c("#D18F55", "#FF3F38","#FF8C8D", "#AFDE62","#3CBC38","#4F5791", "#132157"),c(2,4,3,5,6,1,7)),
  MelonPomelo = rbind(c("#EE404E","#F99D53","#F9E7C2","#24C852","#4F5791", "#132157"), c(1,4,2,5,3,6)),
  MurePepino = rbind(c("#D64358","#EAFB88","#3C8C4D","#DFCEE0","#4F5791", "#132157"),c(1,3,2,4,5,6)),

#Paired Palettes
#lacroix_palette(type = "paired")
paired = c("#C70E7B", "#FC6882","#007BC3","#54BCD1","#EF7C12","#F4B95A","#009F3F","#8FDA04",
             "#AF6125","#F4E3C7","#B25D91","#EFC7E6", "#8F92A1", "#AFDFEF", "#172869")  
```

##################################################################################
##################################################################################
##                                 Checkpoint                                   ##
##################################################################################
##################################################################################

#Good chance to save the phyloseq object and space without any analyses run.
```{r}
#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_diel_preanalysischeckpoint.RData")
```

#Bring the environment back to this point to move forward
```{r}
load("/Users/bradleyweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_diel_preanalysischeckpoint.RData")



load("/Users/bradleyweiler/My Drive (baw117@miami.edu)/curacao/metabarcoding/data/16s_diel/16s_diel_preanalysischeckpoint.RData")
```



##################################################################################
##################################################################################
##                                     Stats                                    ##
##################################################################################
##################################################################################


##################################################################################
##                                    Alpha                                     ##
##################################################################################

#PSTR  **SHANNON**
```{r}
ad_pstr <- prune_taxa(taxa_sums(diel_pstr) > 0, diel_pstr)
tab.shan <- microbiome::alpha(ad_pstr, index = "diversity_shannon")
kable(head(tab.shan))

ps1.meta <- meta(ad_pstr)
kable(head(ps1.meta))
ps1.meta$Shannon <- tab.shan$diversity_shannon 

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_dieltime_shan <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Shannon)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_pvalue_manual(Shannon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylim(0,6) +
                            ylab("Shannon Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1)) #+
                            #stat_compare_means(comparisons = fam.pairs) 
                            
                        

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/alpha/pstr_dieltime_alpha_shannon_clean_ylim.pdf", width = 6, height = 6)
pstr_dieltime_shan
dev.off()
```

#PSTR CHLORO ONLY **SHANNON**
```{r}

ad_pstr_plas <- prune_taxa(taxa_sums(diel_pstr_plas) > 0, diel_pstr_plas)
tab.shan <- microbiome::alpha(ad_pstr_plas, index = "diversity_shannon")
kable(head(tab.shan))

ps1.meta <- meta(ad_pstr_plas)
kable(head(ps1.meta))
ps1.meta$Shannon <- tab.shan$diversity_shannon 

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_plas_dieltime_shan <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Shannon)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_pvalue_manual(Shannon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylab("Shannon Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1)) #+
                            #stat_compare_means(comparisons = fam.pairs) 
                            
                        

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/alpha/pstr_plas_dieltime_alpha_shannon_clean.pdf", width = 6, height = 6)
pstr_plas_dieltime_shan
dev.off()
```

#PSTR **rich**
```{r}

ad_pstr <- prune_taxa(taxa_sums(diel_pstr) > 0, diel_pstr)
tab.rich <- richness(ad_pstr)
kable(head(tab.rich))

ps1.meta <- meta(ad_pstr)
kable(head(ps1.meta))
ps1.meta$Richness <- tab.rich$observed

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_dieltime_rich <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Richness)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_pvalue_manual(richnon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylim(0,2500) +
                            ylab("Richness Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1)) #+
                            #stat_compare_means(comparisons = fam.pairs) 
                            
                        

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/alpha/pstr_dieltime_alpha_rich_clean_ylim.pdf", width = 6, height = 6)
pstr_dieltime_rich
dev.off()
```

#PSTR CHLORO ONLY **rich**
```{r}

ad_pstr_plas <- prune_taxa(taxa_sums(diel_pstr_plas) > 0, diel_pstr_plas)
tab.rich <- richness(ad_pstr_plas)
kable(head(tab.rich))

ps1.meta <- meta(ad_pstr_plas)
kable(head(ps1.meta))
ps1.meta$Richness <- tab.rich$observed 

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_plas_dieltime_rich <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Richness)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_pvalue_manual(richnon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylab("Richness Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1)) #+
                            #stat_compare_means(comparisons = fam.pairs) 
                            
                        

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/alpha/pstr_plas_dieltime_alpha_rich_clean.pdf", width = 6, height = 6)
pstr_plas_dieltime_rich
dev.off()

```




#TimePoints **rich**
```{r}

ad_pstr <- prune_taxa(taxa_sums(pstr6p) > 0, pstr6p)
tab.rich <- richness(ad_pstr)
kable(head(tab.rich))

ps1.meta <- meta(ad_pstr)
kable(head(ps1.meta))
ps1.meta$Richness <- tab.rich$observed

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_day) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_dieltime_rich <- ggplot(ps1.meta, aes(x = factor(diel_day), y = Richness)) +
                            geom_violinhalf(mapping = aes(fill = diel_day), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            #scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_day),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_pvalue_manual(richnon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylim(0,2500) +
                            ylab("Richness Diversity") +
                            xlab("Day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1))#+
                            #stat_compare_means(comparisons = fam.pairs) 
                            
                        

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/alpha/pstr_alpha_rich_6p_day.pdf", width = 6, height = 6)
pstr_dieltime_rich
dev.off()
```





#PSTR **SHANNON** **WATER**
```{r}

ad_pstr <- prune_taxa(taxa_sums(diel_w) > 0, diel_w)
tab.shan <- microbiome::alpha(ad_pstr, index = "diversity_shannon")
kable(head(tab.shan))

ps1.meta <- meta(ad_pstr)
kable(head(ps1.meta))
ps1.meta$Shannon <- tab.shan$diversity_shannon 

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_dieltime_shan <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Shannon)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_compare_means(comparisons = fam.pairs) +
                            #stat_pvalue_manual(Shannon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylim(0,10) +
                            ylab("Shannon Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1)) 
                            
                        

pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/alpha/pstr_dieltime_alpha_shannon_ylim_water.pdf", width = 6, height = 6)
pstr_dieltime_shan
dev.off()
```


#PSTR **rich** **WATER**
```{r}

ad_pstr <- prune_taxa(taxa_sums(diel_w) > 0, diel_w)
tab.rich <- richness(ad_pstr)
kable(head(tab.rich))

ps1.meta <- meta(ad_pstr)
kable(head(ps1.meta))
ps1.meta$Richness <- tab.rich$observed

# create a list of pairwise comparisons
fam <- unique(ps1.meta$diel_time) # get the variables

# make a pairwise list that we want to compare.
fam.pairs <- combn(seq_along(fam), 2, simplify = FALSE, FUN = function(i)fam[i])

print(fam.pairs)

pstr_dieltime_rich <- ggplot(ps1.meta, aes(x = factor(diel_time, levels= timeorder), y = Richness)) +
                            geom_violinhalf(mapping = aes(fill = diel_time), scale = "width", width = 0.6, flip= TRUE, trim = FALSE) +
                            scale_fill_manual(values = diel_palette_ordered) +
                            scale_x_discrete(breaks=timeorder) +
                            geom_half_boxplot(mapping = aes(fill = diel_time),
                                              outlier.size = 0.01, width = 0.3, size = 0.5, alpha = 0.4, side = "r") +
                            #stat_compare_means(comparisons = fam.pairs) +
                            #stat_pvalue_manual(richnon, hide.ns = F, y.position = y.position, label = "FDR.signif", 
                            #                   size = p.val.size, tip.length = 0) +
                            theme_classic() +
                            ylim(0,6000) +
                            ylab("Richness Diversity") +
                            xlab("Time of day") +
                            theme(axis.text.x = element_text(angle = 45, hjust = 1))
                            
                        

pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/alpha/pstr_dieltime_alpha_rich_ylim_water.pdf", width = 6, height = 6)
pstr_dieltime_rich
dev.off()
```


##################################################################################
##################################################################################
##                                 Aitchison PCA                                ##
##################################################################################
##################################################################################


#################################################################################
##############                        PSTR                         ##############
#################################################################################

#PCA 
```{r}
pstr_clr <- microbiome::transform(diel_pstr, 'clr', shift = 1)

pstr_clr.ord <- ordinate(pstr_clr, "RDA", "euclidean")

#pca_pstr_title <- expression(paste("16s Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_pstr_diel = plot_ordination(pstr_clr, pstr_clr.ord, 
                                    shape=1,
                                    color="diel_time"
                                    ) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette_ordered) + 
                                    stat_ellipse(lwd=1)

pca_pstr_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/pstr_pca_dieltime_clean.pdf", width = 6, height = 6)
pca_pstr_diel
dev.off()

library(vegan)
library(phyloseq)

# 1. Extract CLR matrix (samples x features)
clr_mat <- as.data.frame(t(otu_table(pstr_clr)))

# 2. Compute Euclidean distance (Aitchison after CLR)
clr_dist <- dist(clr_mat, method = "euclidean")

# 3. Get sample metadata as a clean data frame
meta <- data.frame(sample_data(pstr_clr))
meta$diel_time <- as.factor(meta$diel_time)  # ensure it's a factor

# 4. PERMANOVA
set.seed(123)
adonis_res <- adonis2(clr_dist ~ diel_time, data = meta, permutations = 999)
print(adonis_res)
```


#PCA **WATER**
```{r}
pstr_clr <- microbiome::transform(diel_w, 'clr', shift = 1)

pstr_clr.ord <- ordinate(pstr_clr, "RDA", "euclidean")

#pca_pstr_title <- expression(paste("18S Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_pstr_diel = plot_ordination(pstr_clr, pstr_clr.ord, 
                                    shape=1,
                                    color="diel_time"#,
                                    #title=pca_pstr_title
                                    ) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    #scale_colour_manual(values=diel_palette_ordered) + 
                                    stat_ellipse(aes(color = diel_day), lwd = 1)

pca_pstr_diel

pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/beta/pstr_pca_dieltime_water_ellipses.pdf", width = 6, height = 6)
pca_pstr_diel
dev.off()

# 1. Extract CLR matrix (samples x features)
clr_mat <- as.data.frame(t(otu_table(pstr_clr)))

# 2. Compute Euclidean distance (Aitchison after CLR)
clr_dist <- dist(clr_mat, method = "euclidean")

# 3. Get sample metadata as a clean data frame
meta <- data.frame(sample_data(pstr_clr))
meta$diel_time <- as.factor(meta$diel_time)  # ensure it's a factor

# 4. PERMANOVA
set.seed(123)
adonis_res <- adonis2(clr_dist ~ diel_day, data = meta, permutations = 999)
print(adonis_res)
```


#PCA w/o SYMBIONTS **OTHER VARIABLES**
```{r}
pstr_clr <- microbiome::transform(diel_pstr, 'clr', shift = 1)

pstr_clr.ord <- ordinate(pstr_clr, "RDA", "euclidean")

#pca_pstr_title <- expression(paste("16s Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_pstr_diel = plot_ordination(pstr_clr, pstr_clr.ord, 
                                    shape=1,
                                    color="diel_day",
                                    title=pca_pstr_title) + 
                                    theme_classic() + 
                                    #geom_point(aes(color = c("midnightblue", "gold")), alpha = 0.8, size = 3) + 
                                    #scale_colour_manual(values=diel_palette_ordered) + 
                                    stat_ellipse(lwd=1)

pca_pstr_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/pstr_pca_variable_clean.pdf", width = 6, height = 6)
pca_pstr_diel
dev.off()
```

#PCA PSTR PLAS
```{r}
pstr_plas_clr <- microbiome::transform(diel_pstr_plas, 'clr', shift = 1)

pstr_plas_clr.ord <- ordinate(pstr_plas_clr, "RDA", "euclidean")

pca_pstr_plas_title <- expression(paste("Plastid Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_pstr_plas_diel = plot_ordination(pstr_clr, pstr_plas_clr.ord, 
                                    shape=1,
                                    color="diel_time",
                                    title=pca_pstr_plas_title) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette_ordered) + 
                                    stat_ellipse(lwd=1)

pca_pstr_plas_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/pstr_plas_pca_dieltime_clean.pdf", width = 6, height = 6)
pca_pstr_plas_diel
dev.off()
```


#################################################################################
##############                        OANN                         ##############
#################################################################################

#PCA w/o SYMBIONTS
```{r}
oann_clr <- microbiome::transform(diel_oann, 'clr', shift = 1)

oann_clr.ord <- ordinate(oann_clr, "RDA", "euclidean")

pca_oann_title <- expression(paste("16s Aitchison distance PCA for ", italic("O. annularis "), "through time"))

pca_oann_diel = plot_ordination(oann_clr, oann_clr.ord, 
                                    shape=1,
                                    color="diel_time",
                                    title=pca_oann_title) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette) + 
                                    stat_ellipse(lwd=1)

pca_oann_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/oann_pca_dieltime.pdf", width = 12, height = 8)
pca_oann_diel
dev.off()
```

#PCA SYMBIONTS
```{r}
oann_clr <- microbiome::transform(dielsymb_oann, 'clr', shift = 1)

oann_clr.ord <- ordinate(oann_clr, "RDA", "euclidean")

pca_oann_title <- expression(paste("Symbiont Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_oann_diel = plot_ordination(oann_clr, oann_clr.ord, 
                                    shape=1,
                                    color="diel_time",
                                    title=pca_oann_title) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette) + 
                                    stat_ellipse(lwd=1)

pca_oann_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/oann_symb_pca_dieltime.pdf", width = 12, height = 8)
pca_oann_diel
dev.off()
```

#################################################################################
##############                        DLAB                         ##############
#################################################################################

#PCA w/o SYMBIONTS
```{r}
dlab_clr <- microbiome::transform(diel_dlab, 'clr', shift = 1)

dlab_clr.ord <- ordinate(dlab_clr, "RDA", "euclidean")

pca_dlab_title <- expression(paste("16s Aitchison distance PCA for ", italic("D. labyrinthiformes "), "through time"))

pca_dlab_diel = plot_ordination(dlab_clr, dlab_clr.ord, 
                                    shape=1,
                                    color="diel_time",
                                    title=pca_dlab_title) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette) + 
                                    stat_ellipse(lwd=1)

pca_dlab_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/dlab_pca_dieltime.pdf", width = 12, height = 8)
pca_dlab_diel
dev.off()
```

#PCA SYMBIONTS
```{r}
dlab_clr <- microbiome::transform(dielsymb_dlab, 'clr', shift = 1)

dlab_clr.ord <- ordinate(dlab_clr, "RDA", "euclidean")

pca_dlab_title <- expression(paste("Symbiont Aitchison distance PCA for ", italic("P. strigosa "), "through time"))

pca_dlab_diel = plot_ordination(dlab_clr, dlab_clr.ord, 
                                    shape=1,
                                    color="diel_time",
                                    title=pca_dlab_title) + 
                                    theme_classic() + 
                                    geom_point(aes(color = diel_time), alpha = 0.8, size = 3) + 
                                    scale_colour_manual(values=diel_palette) + 
                                    stat_ellipse(lwd=1)

pca_dlab_diel

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/beta/dlab_symb_pca_dieltime.pdf", width = 12, height = 8)
pca_dlab_diel
dev.off()
```



##################################################################################
#######                             Bubble Plots                            ######
#######                            (Standard Dev)                           ######
##################################################################################

#################################################################################
##############                        PSTR                         ##############
#################################################################################

# PSTR by diel time w/ STANDARD DEV **ORDER LEVEL**
```{r}
sd_diel_pstr <- diel_pstr %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr <- merge_samples(diel_pstr, diel_pstr@sam_data$diel_time) 
type_taxa_diel_pstr <- bubSD_diel_pstr %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr <- type_taxa_diel_pstr %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr <- data.table(type_taxa_diel_pstr)

dat_diel_pstr$SD <- sd_diel_pstr$sd
# convert Phylum to a character vector from a factor because R
dat_diel_pstr$Order <- as.character(dat_diel_pstr$Order)
# group dataframe by Phylum, calculate median rel. abundance
dat_diel_pstr[, median := median(Abundance, na.rm = FALSE), 
    by = "Order"]
# Change name to remainder of Phylum less than 2%
dat_diel_pstr[(median <= 0.01), Order := "Below Threshold"]
dat_diel_pstr[(Order == "Below Threshold"), Phylum := "Other"]

bubSD_diel_pstr_title <- expression(paste("Microbial Orders within ", italic("P. strigosa")))

bpSD_diel_pstr <- ggplot(dat_diel_pstr[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Order, levels=bactorder))) + geom_point(aes(size = Abundance, fill=Phylum, color=Phylum), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Prokaryotic Order", size = "Relative Abundance %", fill="Phylum") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_pstr_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Phylum, color=Phylum), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Phylum, color=Phylum), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_bpSD_diel_order.pdf", width = 13, height = 18)
bpSD_diel_pstr
dev.off()
```

# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL**
```{r}

sd_diel_pstr <- diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr <- merge_samples(diel_pstr, diel_pstr@sam_data$diel_time) 
type_taxa_diel_pstr <- bubSD_diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr <- type_taxa_diel_pstr %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr <- data.table(type_taxa_diel_pstr)

dat_diel_pstr$SD <- sd_diel_pstr$sd
# convert Phylum to a character vector from a factor because R
dat_diel_pstr$Genus <- as.character(dat_diel_pstr$Genus)
# group dataframe by Phylum, calculate median rel. abundance
dat_diel_pstr[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Phylum less than 2%
dat_diel_pstr[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_pstr[(Genus == "Below Threshold"), Phylum := "Other"]

bubSD_diel_pstr_title <- expression(paste("Prokaryotic Genera within ", italic("P. strigosa")))

bpSD_diel_pstr <- ggplot(dat_diel_pstr[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Phylum, color=Phylum), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Prokaryotic Genera", size = "Relative Abundance %", fill="Phylum") + theme_bw() +  
  #scale_fill_manual(values = eukPhylum_palette, breaks=c('Other', 'Stramenopiles', 'Alveolata', 'Chlorophyta')) +
  #scale_color_manual(values = eukPhylum_palette,breaks=c('Other', 'Stramenopiles', 'Alveolata', 'Chlorophyta')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_pstr_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Phylum, color=Phylum), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Phylum, color=Phylum), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_bpSD_diel_genus.pdf", width = 11, height = 25)
bpSD_diel_pstr
dev.off()
```


# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL** Endozoicomonas only
```{r}

sd_diel_pstr <- diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr <- merge_samples(diel_pstr, diel_pstr@sam_data$diel_time) 
type_taxa_diel_pstr <- bubSD_diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr <- type_taxa_diel_pstr %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr <- data.table(type_taxa_diel_pstr)

dat_diel_pstr$SD <- sd_diel_pstr$sd
# convert Class to a character vector from a factor because R
dat_diel_pstr$Genus <- as.character(dat_diel_pstr$Genus)
# group dataframe by Class, calculate median rel. abundance
dat_diel_pstr[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Class less than 2%
dat_diel_pstr[(median <= 5), Genus := "Below Threshold"]
dat_diel_pstr[(Genus == "Below Threshold"), Class := "Other"]

bubSD_diel_pstr_title <- expression(paste("Endozoicomonas within ", italic("P. strigosa")))

bpSD_diel_pstr <- ggplot(dat_diel_pstr[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Class, color=Class), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Endozoicomonas", size = "Relative Abundance %", fill="Class") + theme_bw() +  
  #scale_fill_manual(values = eukclass_palette, breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) +
  #scale_color_manual(values = eukclass_palette,breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_pstr_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Class, color=Class), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Class, color=Class), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr


pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_bpSD_diel_genus_endozoicomonas.pdf", width = 6, height = 6)
bpSD_diel_pstr
dev.off()
```


ps_diel_archaea
# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL** Endozoicomonas only
```{r}

sd_diel_pstr_archaea <- ps_diel_archaea %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr_archaea <- merge_samples(ps_diel_archaea, ps_diel_archaea@sam_data$diel_time) 
type_taxa_diel_pstr_archaea <- bubSD_diel_pstr_archaea %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr_archaea <- type_taxa_diel_pstr_archaea %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr_archaea <- data.table(type_taxa_diel_pstr_archaea)

dat_diel_pstr_archaea$SD <- sd_diel_pstr_archaea$sd
# convert Class to a character vector from a factor because R
dat_diel_pstr_archaea$Genus <- as.character(dat_diel_pstr_archaea$Genus)
# group dataframe by Class, calculate median rel. abundance
dat_diel_pstr_archaea[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Class less than 2%
dat_diel_pstr_archaea[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_pstr_archaea[(Genus == "Below Threshold"), Class := "Other"]

bubSD_diel_pstr_archaea_title <- expression(paste("Archaeal taxa within ", italic("P. strigosa")))

bpSD_diel_pstr_archaea <- ggplot(dat_diel_pstr_archaea[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Class, color=Class), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Archael Taxa", size = "Relative Abundance %", fill="Class") + theme_bw() +  
  scale_fill_manual(values = safe_colorblind_palette) +
  scale_color_manual(values = safe_colorblind_palette) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_pstr_archaea_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Class, color=Class), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Class, color=Class), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr_archaea


pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_bpSD_diel_archaea.pdf", width = 6, height = 6)
bpSD_diel_pstr_archaea
dev.off()
```

# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL** CLASS Colour **DONE**
```{r}

sd_diel_pstr <- diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr <- merge_samples(diel_pstr, diel_pstr@sam_data$diel_time) 
type_taxa_diel_pstr <- bubSD_diel_pstr %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr <- type_taxa_diel_pstr %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr <- data.table(type_taxa_diel_pstr)

dat_diel_pstr$SD <- sd_diel_pstr$sd
# convert Class to a character vector from a factor because R
dat_diel_pstr$Genus <- as.character(dat_diel_pstr$Genus)
# group dataframe by Class, calculate median rel. abundance
dat_diel_pstr[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Class less than 2%
dat_diel_pstr[(median <= 0.1), Genus := "Below Threshold"]
dat_diel_pstr[(Genus == "Below Threshold"), Class := "Other"]

bubSD_diel_pstr_title <- expression(paste("Prokaryotic Genera within ", italic("P. strigosa")))

bpSD_diel_pstr <- ggplot(dat_diel_pstr[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Class, color=Class), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Prokaryotic Genus (10%)", size = "Relative Abundance %", fill="Class") + theme_bw() +  
  #scale_fill_manual(values = eukclass_palette, breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) +
  #scale_color_manual(values = eukclass_palette,breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_pstr_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Class, color=Class), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Class, color=Class), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr


pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_bpSD_diel_genus_10perc_clean_newtax.pdf", width = 6, height = 12)
bpSD_diel_pstr
dev.off()
```


# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL** CLASS Colour **DONE** **WATER**
```{r}

sd_diel_w <- diel_w %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_w <- merge_samples(diel_w, diel_w@sam_data$diel_time) 
type_taxa_diel_w <- bubSD_diel_w %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_w <- type_taxa_diel_w %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_w <- data.table(type_taxa_diel_w)

dat_diel_w$SD <- sd_diel_w$sd
# convert Class to a character vector from a factor because R
dat_diel_w$Genus <- as.character(dat_diel_w$Genus)
# group dataframe by Class, calculate median rel. abundance
dat_diel_w[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Class less than 2%
dat_diel_w[(median <= 1), Genus := "Below Threshold"]
dat_diel_w[(Genus == "Below Threshold"), Class := "Other"]

bubSD_diel_w_title <- expression(paste("Prokaryotic Genera within ", italic("P. strigosa")))

bpSD_diel_w <- ggplot(dat_diel_w[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Class, color=Class), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Prokaryotic Genus (1%)", size = "Relative Abundance %", fill="Class") + theme_bw() +  
  #scale_fill_manual(values = eukclass_palette, breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) +
  #scale_color_manual(values = eukclass_palette,breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_w_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Class, color=Class), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Class, color=Class), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_w


pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/community/pstr_bpSD_diel_genus_water.pdf", width = 6, height = 6)
bpSD_diel_w
dev.off()
```


# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL** CLASS Colour **DONE** **WATER**
```{r}
#ps_diel_archaea<- subset_taxa(ps_diel_filtered, (Kingdom=="Archaea" | is.na(Kingdom)))
diel_w_arch <- subset_taxa(diel_w, (Kingdom=="Archaea" | is.na(Kingdom)))

sd_diel_w_arch <- diel_w_arch %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_w_arch <- merge_samples(diel_w_arch, diel_w_arch@sam_data$diel_time) 
type_taxa_diel_w_arch <- bubSD_diel_w_arch %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_w_arch <- type_taxa_diel_w_arch %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_w_arch <- data.table(type_taxa_diel_w_arch)

dat_diel_w_arch$SD <- sd_diel_w_arch$sd
# convert Class to a character vector from a factor because R
dat_diel_w_arch$Genus <- as.character(dat_diel_w_arch$Genus)
# group dataframe by Class, calculate median rel. abundance
dat_diel_w_arch[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Class less than 2%
dat_diel_w_arch[(median <= 0.001), Genus := "Below Threshold"]
dat_diel_w_arch[(Genus == "Below Threshold"), Class := "Other"]

bubSD_diel_w_arch_title <- expression(paste("Prokaryotic Genera within ", italic("P. strigosa")))

bpSD_diel_w_arch <- ggplot(dat_diel_w_arch[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=bactgenus))) + geom_point(aes(size = Abundance, fill=Class, color=Class), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Prokaryotic Genus (1%)", size = "Relative Abundance %", fill="Class") + theme_bw() +  
  #scale_fill_manual(values = eukclass_palette, breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) +
  #scale_color_manual(values = eukclass_palette,breaks=c('Other', 'Bacillariophyceae', 'Syndiniales', 'Dinophyceae',   'Oligohymenophorea', 'Coccidiomorphea', 'Ulvophyceae')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  #ggtitle(label=bubSD_diel_w_arch_title, subtitle = "With median relative abundance above 1%") + 
  geom_point(aes(size=Abundance+SD, fill = Class, color=Class), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Class, color=Class), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_w_arch


pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/community/pstr_bpSD_diel_water_arch_0.001per.pdf", width = 6, height = 3)
bpSD_diel_w_arch
dev.off()
```




# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL PLAS**
```{r}
sd_diel_pstr_plas <- diel_pstr_plas %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr_plas <- merge_samples(diel_pstr_plas, diel_pstr_plas@sam_data$diel_time) 
type_taxa_diel_pstr_plas <- bubSD_diel_pstr_plas %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr_plas <- type_taxa_diel_pstr_plas %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr_plas <- data.table(type_taxa_diel_pstr_plas)

dat_diel_pstr_plas$SD <- sd_diel_pstr_plas$sd
# convert Subdivision to a character vector from a factor because R
dat_diel_pstr_plas$Genus <- as.character(dat_diel_pstr_plas$Genus)
# group dataframe by Subdivision, calculate median rel. abundance
dat_diel_pstr_plas[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Subdivision less than 2%
dat_diel_pstr_plas[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_pstr_plas[(Genus == "Below Threshold"), Subdivision := "Other"]

bubSD_diel_pstr_plas_title <- expression(paste("Genus of plastid bearing eukaryotes within ", italic("P. strigosa")))

bpSD_diel_pstr_plas <- ggplot(dat_diel_pstr_plas[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=plast_genus))) + geom_point(aes(size = Abundance, fill=Subdivision, color=Subdivision), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Genus", size = "Relative Abundance %", fill="Subdivision") + theme_bw() +  
  scale_fill_manual(values = plassubdiv_palette, breaks=c('Other', 'Gyrista:plas', 'Chrompodellids:plas', 'Haptophyta_X:plas', 'Chlorophyta_X:plas')) +
  scale_color_manual(values = plassubdiv_palette,breaks=c('Other', 'Gyrista:plas', 'Chrompodellids:plas', 'Haptophyta_X:plas', 'Chlorophyta_X:plas')) +  
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_pstr_plas_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Subdivision, color=Subdivision), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Subdivision, color=Subdivision), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr_plas

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_plas_bpSD_diel_genus_Subdivision.pdf", width = 8, height = 5)
bpSD_diel_pstr_plas
dev.off()
```
# PSTR by diel time w/ STANDARD DEV **GENUS LEVEL PLAS**
```{r}
sd_diel_pstr_plas <- diel_pstr_plas %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr_plas <- merge_samples(diel_pstr_plas, diel_pstr_plas@sam_data$diel_time) 
type_taxa_diel_pstr_plas <- bubSD_diel_pstr_plas %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr_plas <- type_taxa_diel_pstr_plas %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr_plas <- data.table(type_taxa_diel_pstr_plas)

dat_diel_pstr_plas$SD <- sd_diel_pstr_plas$sd
# convert Subdivision to a character vector from a factor because R
dat_diel_pstr_plas$Genus <- as.character(dat_diel_pstr_plas$Genus)
# group dataframe by Subdivision, calculate median rel. abundance
dat_diel_pstr_plas[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Subdivision less than 2%
dat_diel_pstr_plas[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_pstr_plas[(Genus == "Below Threshold"), Subdivision := "Other"]

bubSD_diel_pstr_plas_title <- expression(paste("Genus of plastid bearing eukaryotes within ", italic("P. strigosa")))

bpSD_diel_pstr_plas <- ggplot(dat_diel_pstr_plas[Abundance > 0], aes(x = factor(Sample, levels=c("T6P","T12P","T6A","T12A")), y = factor(Genus, levels=plast_genus))) + geom_point(aes(size = Abundance, fill=Subdivision, color=Subdivision), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Genus", size = "Relative Abundance %", fill="Subdivision") + theme_bw() +  
  scale_fill_manual(values = plassubdiv_palette, breaks=c('Other', 'Gyrista:plas', 'Chrompodellids:plas', 'Haptophyta_X:plas', 'Chlorophyta_X:plas')) +
  scale_color_manual(values = plassubdiv_palette,breaks=c('Other', 'Gyrista:plas', 'Chrompodellids:plas', 'Haptophyta_X:plas', 'Chlorophyta_X:plas')) +  
  theme(axis.text.x = element_text(angle = 45, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_pstr_plas_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Subdivision, color=Subdivision), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Subdivision, color=Subdivision), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr_plas + 
  coord_flip()

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_plas_bpSD_diel_genus_Subdivision.pdf", width = 8, height = 4)
bpSD_diel_pstr_plas+ coord_flip()
dev.off()
```


# PSTR by diel time w/ STANDARD DEV **SPECIES LEVEL PLAS** SubDivision Colour
```{r}
sd_diel_pstr_plas <- diel_pstr_plas %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr_plas <- merge_samples(diel_pstr_plas, diel_pstr_plas@sam_data$diel_time) 
type_taxa_diel_pstr_plas <- bubSD_diel_pstr_plas %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr_plas <- type_taxa_diel_pstr_plas %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr_plas <- data.table(type_taxa_diel_pstr_plas)

dat_diel_pstr_plas$SD <- sd_diel_pstr_plas$sd
# convert Subdivision to a character vector from a factor because R
dat_diel_pstr_plas$Species <- as.character(dat_diel_pstr_plas$Species)
# group dataframe by Subdivision, calculate median rel. abundance
dat_diel_pstr_plas[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Subdivision less than 2%
dat_diel_pstr_plas[(median <= 0.01), Species := "Below Threshold"]
dat_diel_pstr_plas[(Species == "Below Threshold"), Subdivision := "Other"]

bubSD_diel_pstr_plas_title <- expression(paste("Species of plastid bearing eukaryotes within ", italic("P. strigosa")))

bpSD_diel_pstr_plas <- ggplot(dat_diel_pstr_plas[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=plast_species))) + geom_point(aes(size = Abundance, fill=Subdivision, color=Subdivision), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Species", size = "Relative Abundance %", fill="Subdivision") + theme_bw() +  
  #scale_fill_manual(values = euksymbSubdivision_palette, breaks=c('Other', 'Durusdinium', 'Cladocopium', 'Breviolum', 'Symbiodinium')) +
  #scale_color_manual(values = euksymbSubdivision_palette,breaks=c('Other', 'Durusdinium', 'Cladocopium', 'Breviolum', 'Symbiodinium')) +  
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_pstr_plas_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Subdivision, color=Subdivision), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Subdivision, color=Subdivision), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr_plas

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_plas_bpSD_diel_species_subdivision.pdf", width = 6, height = 7)
bpSD_diel_pstr_plas
dev.off()
```
# PSTR by diel time w/ STANDARD DEV **SPECIES LEVEL PLAS** genus colour
```{r}
sd_diel_pstr_plas <- diel_pstr_plas %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_pstr_plas <- merge_samples(diel_pstr_plas, diel_pstr_plas@sam_data$diel_time) 
type_taxa_diel_pstr_plas <- bubSD_diel_pstr_plas %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_pstr_plas <- type_taxa_diel_pstr_plas %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_pstr_plas <- data.table(type_taxa_diel_pstr_plas)

dat_diel_pstr_plas$SD <- sd_diel_pstr_plas$sd
# convert Genus to a character vector from a factor because R
dat_diel_pstr_plas$Species <- as.character(dat_diel_pstr_plas$Species)
# group dataframe by Genus, calculate median rel. abundance
dat_diel_pstr_plas[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Genus less than 2%
dat_diel_pstr_plas[(median <= 0.01), Species := "Below Threshold"]
dat_diel_pstr_plas[(Species == "Below Threshold"), Genus := "Other"]

bubSD_diel_pstr_plas_title <- expression(paste("Species of plastid bearing eukaryotes within ", italic("P. strigosa")))

bpSD_diel_pstr_plas <- ggplot(dat_diel_pstr_plas[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=plast_species))) + geom_point(aes(size = Abundance, fill=Genus, color=Genus), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Species", size = "Relative Abundance %", fill="Genus") + theme_bw() +  
  #scale_fill_manual(values = euksymbgenus_palette, breaks=c('Other', 'Durusdinium', 'Cladocopium', 'Breviolum', 'Symbiodinium')) +
  #scale_color_manual(values = euksymbgenus_palette,breaks=c('Other', 'Durusdinium', 'Cladocopium', 'Breviolum', 'Symbiodinium')) +  
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_pstr_plas_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Genus, color=Genus), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Genus, color=Genus), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_pstr_plas

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/pstr_plas_bpSD_diel_species.pdf", width = 8, height = 8)
bpSD_diel_pstr_plas
dev.off()
```

#################################################################################
##############                        OANN                         ##############
#################################################################################

# oann by diel time w/ STANDARD DEV **ORDER LEVEL W/O SYMBIODINIACEAE**
```{r}
sd_diel_oann <- diel_oann %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_oann <- merge_samples(diel_oann, diel_oann@sam_data$diel_time) 
type_taxa_diel_oann <- bubSD_diel_oann %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_oann <- type_taxa_diel_oann %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_oann <- data.table(type_taxa_diel_oann)

dat_diel_oann$SD <- sd_diel_oann$sd
# convert Division to a character vector from a factor because R
dat_diel_oann$Order <- as.character(dat_diel_oann$Order)
# group dataframe by Division, calculate median rel. abundance
dat_diel_oann[, median := median(Abundance, na.rm = FALSE), 
    by = "Order"]
# Change name to remainder of Division less than 2%
dat_diel_oann[(median <= 0.01), Order := "Below Threshold"]
dat_diel_oann[(Order == "Below Threshold"), Division := "Other"]

bubSD_diel_oann_title <- expression(paste("Eukaryotic Orders within ", italic("O. annularis")))

bpSD_diel_oann <- ggplot(dat_diel_oann[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Order, levels=eukorder))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Order (Excluding Symbiodiniaceae)", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_oann_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/oann_bpSD_diel_order.pdf", width = 13, height = 12)
bpSD_diel_oann
dev.off()
```

# oann by diel time w/ STANDARD DEV **GENUS LEVEL W/O SYMBIODINIACEAE**
```{r}

sd_diel_oann <- diel_oann %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_oann <- merge_samples(diel_oann, diel_oann@sam_data$diel_time) 
type_taxa_diel_oann <- bubSD_diel_oann %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_oann <- type_taxa_diel_oann %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_oann <- data.table(type_taxa_diel_oann)

dat_diel_oann$SD <- sd_diel_oann$sd
# convert Division to a character vector from a factor because R
dat_diel_oann$Genus <- as.character(dat_diel_oann$Genus)
# group dataframe by Division, calculate median rel. abundance
dat_diel_oann[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Division less than 2%
dat_diel_oann[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_oann[(Genus == "Below Threshold"), Division := "Other"]

bubSD_diel_oann_title <- expression(paste("Eukaryotic Genera (Excluding Symbiodiniaceae) within ", italic("O. annularis")))

bpSD_diel_oann <- ggplot(dat_diel_oann[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=eukgenus))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Genus", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_oann_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/oann_bpSD_diel_genus.pdf", width = 13, height = 12)
bpSD_diel_oann
dev.off()
```

# oann by diel time w/ STANDARD DEV **SPECIES LEVEL W/O SYMBIODINIACEAE**
```{r}

sd_diel_oann <- diel_oann %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_oann <- merge_samples(diel_oann, diel_oann@sam_data$diel_time) 
type_taxa_diel_oann <- bubSD_diel_oann %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_oann <- type_taxa_diel_oann %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_oann <- data.table(type_taxa_diel_oann)

dat_diel_oann$SD <- sd_diel_oann$sd
# convert Division to a character vector from a factor because R
dat_diel_oann$Species <- as.character(dat_diel_oann$Species)
# group dataframe by Division, calculate median rel. abundance
dat_diel_oann[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Division less than 2%
dat_diel_oann[(median <= 0.01), Species := "Below Threshold"]
dat_diel_oann[(Species == "Below Threshold"), Division := "Other"]

bubSD_diel_oann_title <- expression(paste("Eukaryotic Species (Excluding Symbiodiniaceae) within ", italic("O. annularis")))

bpSD_diel_oann <- ggplot(dat_diel_oann[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=eukspecies))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Species", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_oann_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/oann_bpSD_diel_species.pdf", width = 13, height = 12)
bpSD_diel_oann
dev.off()
```

# oann by diel time w/ STANDARD DEV **SPECIES LEVEL SYMBIODINIACEAE**
```{r}
sd_dielsymb_oann <- dielsymb_oann %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_dielsymb_oann <- merge_samples(dielsymb_oann, dielsymb_oann@sam_data$diel_time) 
type_taxa_dielsymb_oann <- bubSD_dielsymb_oann %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_dielsymb_oann <- type_taxa_dielsymb_oann %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_dielsymb_oann <- data.table(type_taxa_dielsymb_oann)

dat_dielsymb_oann$SD <- sd_dielsymb_oann$sd
# convert Genus to a character vector from a factor because R
dat_dielsymb_oann$Species <- as.character(dat_dielsymb_oann$Species)
# group dataframe by Genus, calculate median rel. abundance
dat_dielsymb_oann[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Genus less than 2%
dat_dielsymb_oann[(median <= 0.01), Species := "Below Threshold"]
dat_dielsymb_oann[(Species == "Below Threshold"), Genus := "Other"]

bubSD_dielsymb_oann_title <- expression(paste("Species of Symbiodiniaceae within ", italic("O. annularis")))

bpSD_dielsymb_oann <- ggplot(dat_dielsymb_oann[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=eukspecies))) + geom_point(aes(size = Abundance, fill=Genus, color=Genus), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Species", size = "Relative Abundance %", fill="Genus") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_dielsymb_oann_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Genus, color=Genus), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Genus, color=Genus), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_dielsymb_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/oann_symb_bpSD_diel_species.pdf", width = 13, height = 12)
bpSD_dielsymb_oann
dev.off()
```



#################################################################################
##############                        DLAB                         ##############
#################################################################################

# dlab by diel time w/ STANDARD DEV **ORDER LEVEL W/O SYMBIODINIACEAE**
```{r}
sd_diel_dlab <- diel_dlab %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_dlab <- merge_samples(diel_dlab, diel_dlab@sam_data$diel_time) 
type_taxa_diel_dlab <- bubSD_diel_dlab %>% 
  tax_glom(taxrank = "Order") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_dlab <- type_taxa_diel_dlab %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_dlab <- data.table(type_taxa_diel_dlab)

dat_diel_dlab$SD <- sd_diel_dlab$sd
# convert Division to a character vector from a factor because R
dat_diel_dlab$Order <- as.character(dat_diel_dlab$Order)
# group dataframe by Division, calculate median rel. abundance
dat_diel_dlab[, median := median(Abundance, na.rm = FALSE), 
    by = "Order"]
# Change name to remainder of Division less than 2%
dat_diel_dlab[(median <= 0.01), Order := "Below Threshold"]
dat_diel_dlab[(Order == "Below Threshold"), Division := "Other"]

bubSD_diel_dlab_title <- expression(paste("Eukaryotic Orders within ", italic("D. labyrinthiformes")))

bpSD_diel_dlab <- ggplot(dat_diel_dlab[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Order, levels=eukorder))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Order (Excluding Symbiodiniaceae)", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_dlab_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/dlab_bpSD_diel_order.pdf", width = 13, height = 12)
bpSD_diel_dlab
dev.off()
```

# dlab by diel time w/ STANDARD DEV **GENUS LEVEL W/O SYMBIODINIACEAE**
```{r}

sd_diel_dlab <- diel_dlab %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_dlab <- merge_samples(diel_dlab, diel_dlab@sam_data$diel_time) 
type_taxa_diel_dlab <- bubSD_diel_dlab %>% 
  tax_glom(taxrank = "Genus") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_dlab <- type_taxa_diel_dlab %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_dlab <- data.table(type_taxa_diel_dlab)

dat_diel_dlab$SD <- sd_diel_dlab$sd
# convert Division to a character vector from a factor because R
dat_diel_dlab$Genus <- as.character(dat_diel_dlab$Genus)
# group dataframe by Division, calculate median rel. abundance
dat_diel_dlab[, median := median(Abundance, na.rm = FALSE), 
    by = "Genus"]
# Change name to remainder of Division less than 2%
dat_diel_dlab[(median <= 0.01), Genus := "Below Threshold"]
dat_diel_dlab[(Genus == "Below Threshold"), Division := "Other"]

bubSD_diel_dlab_title <- expression(paste("Eukaryotic Genera (Excluding Symbiodiniaceae) within ", italic("D. labyrinthiformes")))

bpSD_diel_dlab <- ggplot(dat_diel_dlab[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Genus, levels=eukgenus))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Genus", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_dlab_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/dlab_bpSD_diel_genus.pdf", width = 13, height = 12)
bpSD_diel_dlab
dev.off()
```

# dlab by diel time w/ STANDARD DEV **SPECIES LEVEL W/O SYMBIODINIACEAE**
```{r}

sd_diel_dlab <- diel_dlab %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_diel_dlab <- merge_samples(diel_dlab, diel_dlab@sam_data$diel_time) 
type_taxa_diel_dlab <- bubSD_diel_dlab %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_diel_dlab <- type_taxa_diel_dlab %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_diel_dlab <- data.table(type_taxa_diel_dlab)

dat_diel_dlab$SD <- sd_diel_dlab$sd
# convert Division to a character vector from a factor because R
dat_diel_dlab$Species <- as.character(dat_diel_dlab$Species)
# group dataframe by Division, calculate median rel. abundance
dat_diel_dlab[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Division less than 2%
dat_diel_dlab[(median <= 0.01), Species := "Below Threshold"]
dat_diel_dlab[(Species == "Below Threshold"), Division := "Other"]

bubSD_diel_dlab_title <- expression(paste("Eukaryotic Species (Excluding Symbiodiniaceae) within ", italic("D. labyrinthiformes")))

bpSD_diel_dlab <- ggplot(dat_diel_dlab[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=eukspecies))) + geom_point(aes(size = Abundance, fill=Division, color=Division), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Eukaryotic Species", size = "Relative Abundance %", fill="Division") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_diel_dlab_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Division, color=Division), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Division, color=Division), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_diel_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/dlab_bpSD_diel_species.pdf", width = 13, height = 12)
bpSD_diel_dlab
dev.off()
```

# dlab by diel time w/ STANDARD DEV **SPECIES LEVEL SYMBIODINIACEAE**
```{r}
sd_dielsymb_dlab <- dielsymb_dlab %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} ) %>% 
  psmelt() %>% group_by(OTU, diel_time)  %>%
  summarize(sd = sd(Abundance, na.rm=FALSE)) %>% arrange(OTU)

bubSD_dielsymb_dlab <- merge_samples(dielsymb_dlab, dielsymb_dlab@sam_data$diel_time) 
type_taxa_dielsymb_dlab <- bubSD_dielsymb_dlab %>% 
  tax_glom(taxrank = "Species") %>% 
  transform_sample_counts(function(x) {x/sum(x)*100} )  

type_taxa_dielsymb_dlab <- type_taxa_dielsymb_dlab %>% 
  psmelt() %>% 
  arrange(OTU) 

library(data.table)
# create dataframe from phyloseq object
dat_dielsymb_dlab <- data.table(type_taxa_dielsymb_dlab)

dat_dielsymb_dlab$SD <- sd_dielsymb_dlab$sd
# convert Genus to a character vector from a factor because R
dat_dielsymb_dlab$Species <- as.character(dat_dielsymb_dlab$Species)
# group dataframe by Genus, calculate median rel. abundance
dat_dielsymb_dlab[, median := median(Abundance, na.rm = FALSE), 
    by = "Species"]
# Change name to remainder of Genus less than 2%
dat_dielsymb_dlab[(median <= 0.01), Species := "Below Threshold"]
dat_dielsymb_dlab[(Species == "Below Threshold"), Genus := "Other"]

bubSD_dielsymb_dlab_title <- expression(paste("Species of Symbiodiniaceae within ", italic("D. labyrinthiformes")))

bpSD_dielsymb_dlab <- ggplot(dat_dielsymb_dlab[Abundance > 0], aes(x = factor(Sample, levels=timeorder), y = factor(Species, levels=eukspecies))) + geom_point(aes(size = Abundance, fill=Genus, color=Genus), alpha = 0.5, shape = 21) + 
  scale_size_continuous(limits = c(0.000001, 100), range = c(0.1,10), breaks = c(1,10,50,75)) + 
  labs( x= "Time of Day", y = "Symbiodiniaceae Species", size = "Relative Abundance %", fill="Genus") + theme_bw() +  
  #scale_fill_manual(values = palette, breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) +
  #scale_color_manual(values = palette,breaks=c('Verrucomicrobiota', 'Proteobacteria', 'Firmicutes', 'Cyanobacteria', 'Bacteroidota', 'Bacteria IS', 'Other')) + 
  theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust=1)) + 
  ggtitle(label=bubSD_dielsymb_dlab_title, subtitle = "With median relative abundance above 1%") + geom_point(aes(size=Abundance+SD, fill = Genus, color=Genus), alpha = 0.20, shape=21) + geom_point(aes(size=Abundance-SD, fill = Genus, color=Genus), alpha = 1, shape=21) + guides(color = "none", fill = guide_legend(override.aes = list(size = 3)))
bpSD_dielsymb_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/community/dlab_symb_bpSD_diel_species.pdf", width = 13, height = 12)
bpSD_dielsymb_dlab
dev.off()
```

##################################################################################
##################################################################################
###                          Chord Plots Community                             ###
##################################################################################
##################################################################################



```{r}
library(microeco)
library(htmlwidgets)
library(circlize)
library(chorddiag)
library(webshot)

meco_diel_pstr <- phyloseq2meco(diel_pstr_network)
meco_diel_pstr

```


```{r}


```






##################################################################################
##################################################################################
###                                   ANCOM-BC                                 ###
##################################################################################
##################################################################################

```{r}
library(ANCOMBC)
library(tidyverse)
library(caret)
library(DT)
```


#PSTR DAY VS NIGHT
```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_day_night <- subset_samples(diel_pstr,  diel_pstr@sam_data$day_night=='day' |
                                             diel_pstr@sam_data$day_night=='night')
set.seed(321)

ancom_pstr_day_night = ancombc2(data = pstr_day_night, assay_name = "counts", tax_level = NULL,
                  fix_formula = "day_night", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 1, pseudo_sens = TRUE,
                  prv_cut = 0.001, lib_cut = 0, s0_perc = 0.05,
                  group = "day_night", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = FALSE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                     B = 100))

res_prim_pstr_day_night = ancom_pstr_day_night$res
res_prim_pstr_day_night
df_res_pstr_day_night = res_prim_pstr_day_night %>% dplyr::filter(`diff_day_nightnight`) %>%
    dplyr::select(taxon, contains("day_night")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_day_night = df_res_pstr_day_night %>% 
    arrange(desc(`lfc_day_nightnight`
)) %>%
    mutate(direct = ifelse(`lfc_day_nightnight`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_day_night$taxon = factor(df_fig_res_pstr_day_night$taxon, levels = df_fig_res_pstr_day_night$taxon)
df_fig_res_pstr_day_night$direct = factor(df_fig_res_pstr_day_night$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_day_night.df = as.data.frame(df_fig_res_pstr_day_night)
write.table(df_fig_res_pstr_day_night.df, file = "/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/ancom/ancom_pstr_day_night.tsv")

#TEST TO SUBSET THE RESULTS DOWN SHOWING ONLY THOSE SIGNIFICANT LFC
df_fig_res_pstr_day_night_filt <- df_fig_res_pstr_day_night %>% filter(abs(lfc_day_nightnight) > 1.5 | lfc_day_nightnight < -1.5)


fig_res_pstr_day_night = df_fig_res_pstr_day_night %>%
    ggplot(aes(x = taxon, y = `lfc_day_nightnight`, fill = `lfc_day_nightnight`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_day_nightnight` - `se_day_nightnight`, ymax = `lfc_day_nightnight` + `se_day_nightnight`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from day to night", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma", direction = -1) + scale_color_viridis_c(option = "plasma", direction = -1) +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_day_night

pdf("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/analysis/pstr_16s_diel/ancom/ancom_pstr_day_night_large.pdf", width = 10, height = 6)
fig_res_pstr_day_night
dev.off()


#New Volcano Plot

df_fig_res_pstr_day_night$taxon <- gsub("_X+", " IS", df_fig_res_pstr_day_night$taxon)
df_fig_res_pstr_day_night$neg_log10_p_value <- -log10(df_fig_res_pstr_day_night$q_day_nightnight)
significance_threshold <- -log10(0.001)
pstr_day_night_volcano <- ggplot(df_fig_res_pstr_day_night, aes(x = `lfc_day_nightnight`, y = neg_log10_p_value)) +
  geom_point(color = 'black') +
  geom_point(data = filter(df_fig_res_pstr_day_night, neg_log10_p_value > significance_threshold), 
             color = 'red', size = 3) +
  #geom_text(data = filter(df_fig_res_pstr_day_night, neg_log10_p_value > significance_threshold), 
  #          aes(label = taxon), hjust = 0, vjust = 0) +
  labs(x = 'Log2 Fold Change', y = '-log10(adjusted p-value)',
       title = 'ANCOM-BC2 Significant Taxa') +
  theme_minimal()
pstr_day_night_volcano

```



#PSTR
```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_12A_6A <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T12A' |
                                             diel_pstr@sam_data$diel_time=='T6A')

ancom_pstr_12A_6A = ancombc2(data = pstr_12A_6A, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                     B = 100))

res_prim_pstr_12A_6A = ancom_pstr_12A_6A$res
res_prim_pstr_12A_6A
df_res_pstr_12A_6A = res_prim_pstr_12A_6A %>% dplyr::filter(`diff_diel_timeT6A`) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_12A_6A = df_res_pstr_12A_6A %>% 
    arrange(desc(`lfc_diel_timeT6A`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT6A`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_12A_6A$taxon = factor(df_fig_res_pstr_12A_6A$taxon, levels = df_fig_res_pstr_12A_6A$taxon)
df_fig_res_pstr_12A_6A$direct = factor(df_fig_res_pstr_12A_6A$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_12A_6A.df = as.data.frame(df_fig_res_pstr_12A_6A)
write.table(df_fig_res_pstr_12A_6A.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12A_6A.tsv")


fig_res_pstr_12A_6A = df_fig_res_pstr_12A_6A %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT6A`, fill = `lfc_diel_timeT6A`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT6A` - `se_diel_timeT6A`, ymax = `lfc_diel_timeT6A` + `se_diel_timeT6A`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T12A to T6A", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_12A_6A

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12A_6A.pdf", width = 13, height = 20)
fig_res_pstr_12A_6A
dev.off()

save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
#load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```

```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_6A_12P <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T6A' |
                                             diel_pstr@sam_data$diel_time=='T12P')

ancom_pstr_6A_12P = ancombc2(data = pstr_6A_12P, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

res_prim_pstr_6A_12P = ancom_pstr_6A_12P$res
res_prim_pstr_6A_12P
df_res_pstr_6A_12P = res_prim_pstr_6A_12P %>% dplyr::filter(diff_diel_timeT6A) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_6A_12P = df_res_pstr_6A_12P %>% 
    arrange(desc(`lfc_diel_timeT6A`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT6A`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_6A_12P$taxon = factor(df_fig_res_pstr_6A_12P$taxon, levels = df_fig_res_pstr_6A_12P$taxon)
df_fig_res_pstr_6A_12P$direct = factor(df_fig_res_pstr_6A_12P$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_6A_12P.df = as.data.frame(df_fig_res_pstr_6A_12P)
write.table(df_fig_res_pstr_6A_12P.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6A_12P.tsv")


fig_res_pstr_6A_12P = df_fig_res_pstr_6A_12P %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT6A`, fill = `lfc_diel_timeT6A`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT6A` - `se_diel_timeT6A`, ymax = `lfc_diel_timeT6A` + `se_diel_timeT6A`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T6A to T12P", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_6A_12P

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6A_12P.pdf", width = 13, height = 20)
fig_res_pstr_6A_12P
dev.off()

#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
#load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```


```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_12P_6P <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T12P' |
                                             diel_pstr@sam_data$diel_time=='T6P')

ancom_pstr_12P_6P = ancombc2(data = pstr_12P_6P, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

res_prim_pstr_12P_6P = ancom_pstr_12P_6P$res
res_prim_pstr_12P_6P
df_res_pstr_12P_6P = res_prim_pstr_12P_6P %>% dplyr::filter(diff_diel_timeT6P) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_12P_6P = df_res_pstr_12P_6P %>% 
    arrange(desc(`lfc_diel_timeT6P`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT6P`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_12P_6P$taxon = factor(df_fig_res_pstr_12P_6P$taxon, levels = df_fig_res_pstr_12P_6P$taxon)
df_fig_res_pstr_12P_6P$direct = factor(df_fig_res_pstr_12P_6P$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_12P_6P.df = as.data.frame(df_fig_res_pstr_12P_6P)
write.table(df_fig_res_pstr_12P_6P.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12P_6P.tsv")


fig_res_pstr_12P_6P = df_fig_res_pstr_12P_6P %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT6P`, fill = `lfc_diel_timeT6P`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT6P` - `se_diel_timeT6P`, ymax = `lfc_diel_timeT6P` + `se_diel_timeT6P`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T6A to T12P", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_12P_6P

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12P_6P.pdf", width = 13, height = 20)
fig_res_pstr_12P_6P
dev.off()

#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
#load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```

```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_6P_12A <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T6P' |
                                             diel_pstr@sam_data$diel_time=='T12A')

ancom_pstr_6P_12A = ancombc2(data = pstr_6P_12A, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

res_prim_pstr_6P_12A = ancom_pstr_6P_12A$res
res_prim_pstr_6P_12A
df_res_pstr_6P_12A = res_prim_pstr_6P_12A %>% dplyr::filter(diff_diel_timeT6P) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_6P_12A = df_res_pstr_6P_12A %>% 
    arrange(desc(`lfc_diel_timeT12A`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT12A`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_6P_12A$taxon = factor(df_fig_res_pstr_6P_12A$taxon, levels = df_fig_res_pstr_6P_12A$taxon)
df_fig_res_pstr_6P_12A$direct = factor(df_fig_res_pstr_6P_12A$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_6P_12A.df = as.data.frame(df_fig_res_pstr_6P_12A)
write.table(df_fig_res_pstr_6P_12A.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6P_12A.tsv")


fig_res_pstr_6P_12A = df_fig_res_pstr_6P_12A %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT12A`, fill = `lfc_diel_timeT12A`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT12A` - `se_diel_timeT12A`, ymax = `lfc_diel_timeT12A` + `se_diel_timeT12A`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T6A to T12P", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_6P_12A

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6P_12A.pdf", width = 13, height = 20)
fig_res_pstr_6P_12A
dev.off()

#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
#load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```


```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_12A_12P <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T12A' |
                                             diel_pstr@sam_data$diel_time=='T12P')

ancom_pstr_12A_12P = ancombc2(data = pstr_12A_12P, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

res_prim_pstr_12A_12P = ancom_pstr_12A_12P$res
res_prim_pstr_12A_12P
df_res_pstr_12A_12P = res_prim_pstr_12A_12P %>% dplyr::filter(diff_diel_timeT12P) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_12A_12P = df_res_pstr_12A_12P %>% 
    arrange(desc(`lfc_diel_timeT12P`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT12P`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_12A_12P$taxon = factor(df_fig_res_pstr_12A_12P$taxon, levels = df_fig_res_pstr_12A_12P$taxon)
df_fig_res_pstr_12A_12P$direct = factor(df_fig_res_pstr_12A_12P$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_12A_12P.df = as.data.frame(df_fig_res_pstr_12A_12P)
write.table(df_fig_res_pstr_12A_12P.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12A_12P.tsv")


fig_res_pstr_12A_12P = df_fig_res_pstr_12A_12P %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT12P`, fill = `lfc_diel_timeT12P`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT12P` - `se_diel_timeT12P`, ymax = `lfc_diel_timeT12P` + `se_diel_timeT12P`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T12A to T12P", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_12A_12P

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_12A_12P.pdf", width = 13, height = 20)
fig_res_pstr_12A_12P
dev.off()

#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
#load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```





```{r}
#Subset the phyloseq object to contain only the samples and variables you want to explore
pstr_6A_6P <- subset_samples(diel_pstr,  diel_pstr@sam_data$diel_time=='T6A' |
                                             diel_pstr@sam_data$diel_time=='T6P')

ancom_pstr_6A_6P = ancombc2(data = pstr_6A_6P, assay_name = "counts", tax_level = NULL,
                  fix_formula = "diel_time", rand_formula = NULL,
                  p_adj_method = "fdr", pseudo = 0, pseudo_sens = TRUE,
                  prv_cut = 0.01, lib_cut = 0, s0_perc = 0.05,
                  group = "diel_time", struc_zero = TRUE, neg_lb = FALSE,
                  alpha = 0.05, n_cl = 2, verbose = TRUE,
                  global = FALSE, pairwise = TRUE, dunnet = FALSE, trend = FALSE, #Change this when more than 1 factor or time points
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "fdr", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, -1, 1),
                                                              nrow = 2, 
                                                              byrow = TRUE),
                                                       matrix(c(-1, 0, 1, -1),
                                                              nrow = 2, 
                                                              byrow = TRUE)),
                                       node = list(2, 2),
                                       solver = "ECOS",
                                       B = 100))

res_prim_pstr_6A_6P = ancom_pstr_6A_6P$res
res_prim_pstr_6A_6P
df_res_pstr_6A_6P = res_prim_pstr_6A_6P %>% dplyr::filter(diff_diel_timeT6P) %>%
    dplyr::select(taxon, contains("diel_time")) #CHANGE ME TO VARIABLE

df_fig_res_pstr_6A_6P = df_res_pstr_6A_6P %>% 
    arrange(desc(`lfc_diel_timeT6P`
)) %>%
    mutate(direct = ifelse(`lfc_diel_timeT6P`
 > 0, "Positive LFC", "Negative LFC"))


df_fig_res_pstr_6A_6P$taxon = factor(df_fig_res_pstr_6A_6P$taxon, levels = df_fig_res_pstr_6A_6P$taxon)
df_fig_res_pstr_6A_6P$direct = factor(df_fig_res_pstr_6A_6P$direct, 
                           levels = c("Positive LFC", "Negative LFC"))
  
df_fig_res_pstr_6A_6P.df = as.data.frame(df_fig_res_pstr_6A_6P)
write.table(df_fig_res_pstr_6A_6P.df, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6A_6P.tsv")


fig_res_pstr_6A_6P = df_fig_res_pstr_6A_6P %>%
    ggplot(aes(x = taxon, y = `lfc_diel_timeT6P`, fill = `lfc_diel_timeT6P`)) + 
    geom_bar(stat = "identity", width = 0.7, color = "black", 
             position = position_dodge(width = 0.4)) +
    geom_errorbar(aes(ymin = `lfc_diel_timeT6P` - `se_diel_timeT6P`, ymax = `lfc_diel_timeT6P` + `se_diel_timeT6P`), 
                  width = 0.2, position = position_dodge(0.05), color = "black") + 
    labs(x = "Prokaryotic Order", y = "Log fold change", 
         title = "LFC from T6A to T6P", subtitle = "ANCOM-BC derived signficant taxa") + scale_fill_viridis_c(option = "plasma") + scale_color_viridis_c(option = "plasma") +
  theme_bw() + 
  theme(legend.position = "none",
        panel.grid.minor.y = element_blank()) + coord_flip()

fig_res_pstr_6A_6P

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/ancom/ancom_pstr_6A_6P.pdf", width = 13, height = 20)
fig_res_pstr_6A_6P
dev.off()

#save.image("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_12A_6A.RData")
load("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/ancom_pstr_rbd_AHvsTR.RData")
```



```{r}

```


##################################################################################
##################################################################################
###                          Heatmaps w/ AMPVIS2                               ###
##################################################################################
##################################################################################

##################################################################################
###                                    PSTR                                    ###
##################################################################################

#PSTR (Samples through time, D1-D2-D3) **USED**
```{r}
library(ampvis2)

ampvis2_pstr<- ampvis2::amp_load(diel_pstr)



av2_pstr<- amp_heatmap(ampvis2_pstr, 
                       order_x_by = dielpstrdayorder, 
                       group_by = "sample_name", 
                       #facet_by = "diel_time",
                       tax_add = c("Class","Family","Genus"), 
                       tax_aggregate = "OTU", 
                       plot_values = FALSE,
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/pstr_hm_samplebytimepoint_clean.pdf", width = 14, height = 6)
av2_pstr
dev.off()
```

#PSTR (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_pstr<- amp_load(diel_pstr)

av2_pstr<- amp_heatmap(ampvis2_pstr, 
                       order_x_by = dielpstrorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class","Family","Genus"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       plot_values = FALSE,
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/pstr_hm_samplebyday.pdf", width = 14, height = 6)
av2_pstr
dev.off()
```

#PSTR (Samples through time, D1-D2-D3) **PLAS ONLY** **USED**
```{r}
library(ampvis2)

ampvis2_pstr_plas<- amp_load(diel_pstr_plas)

av2_pstr_plas<- amp_heatmap(ampvis2_pstr_plas, 
                       order_x_by = dielpstrdayorder, 
                       group_by = "sample_name", 
                       #facet_by = "diel_time",
                       tax_add = c("Class","Family","Genus"), 
                       tax_aggregate = "OTU", 
                       plot_values = FALSE,
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_pstr_plas

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/pstr_plas_hm_samplebytimepoint_clean.pdf", width = 14, height = 6)
av2_pstr_plas
dev.off()
```

#PSTR (Samples through time, D1-D2-D3) **SYMBIODINIACEAE ONLY**
```{r}
library(ampvis2)

ampvis2_pstr<- amp_load(dielsymb_pstr)

av2_pstr<- amp_heatmap(ampvis2_pstr, 
                       order_x_by = dielpstrorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/pstr_symb_hm_samplebyday.pdf", width = 18, height = 13)
av2_pstr
dev.off()
```



##################################################################################
###                                    OANN                                    ###
##################################################################################

#oann (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_oann<- amp_load(diel_oann)

av2_oann<- amp_heatmap(ampvis2_oann, 
                       order_x_by = dieloanndayorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/oann_hm_samplebytimepoint.pdf", width = 18, height = 13)
av2_oann
dev.off()
```

#oann (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_oann<- amp_load(diel_oann)

av2_oann<- amp_heatmap(ampvis2_oann, 
                       order_x_by = dieloannorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/oann_hm_samplebyday.pdf", width = 18, height = 13)
av2_oann
dev.off()
```

#oann (Samples through time, D1-D2-D3) **SYMBIODINIACEAE ONLY**
```{r}
library(ampvis2)

ampvis2_oann<- amp_load(dielsymb_oann)

av2_oann<- amp_heatmap(ampvis2_oann, 
                       order_x_by = dieloanndayorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/oann_symb_hm_samplebytimepoint.pdf", width = 18, height = 13)
av2_oann
dev.off()
```

#oann (Samples through time, D1-D2-D3) **SYMBIODINIACEAE ONLY**
```{r}
library(ampvis2)

ampvis2_oann<- amp_load(dielsymb_oann)

av2_oann<- amp_heatmap(ampvis2_oann, 
                       order_x_by = dieloannorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_oann

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/oann_symb_hm_samplebyday.pdf", width = 18, height = 13)
av2_oann
dev.off()
```

##################################################################################
###                                    DLAB                                    ###
##################################################################################

#dlab (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_dlab<- amp_load(diel_dlab)

av2_dlab<- amp_heatmap(ampvis2_dlab, 
                       order_x_by = dieldlabdayorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/dlab_hm_samplebytimepoint.pdf", width = 18, height = 13)
av2_dlab
dev.off()
```

#dlab (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_dlab<- amp_load(diel_dlab)

av2_dlab<- amp_heatmap(ampvis2_dlab, 
                       order_x_by = dieldlaborder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/dlab_hm_samplebyday.pdf", width = 18, height = 13)
av2_dlab
dev.off()
```

#dlab (Samples through time, D1-D2-D3) **SYMBIODINIACEAE ONLY**
```{r}
library(ampvis2)

ampvis2_dlab<- amp_load(dielsymb_dlab)

av2_dlab<- amp_heatmap(ampvis2_dlab, 
                       order_x_by = dieldlabdayorder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/dlab_symb_hm_samplebytimepoint.pdf", width = 18, height = 13)
av2_dlab
dev.off()
```

#dlab (Samples through time, D1-D2-D3) **SYMBIODINIACEAE ONLY**
```{r}
library(ampvis2)

ampvis2_dlab<- amp_load(dielsymb_dlab)

av2_dlab<- amp_heatmap(ampvis2_dlab, 
                       order_x_by = dieldlaborder, 
                       group_by = "sample_name", 
                       tax_add = c("Class", "Order", "Genus", "Species"), 
                       tax_aggregate = "OTU", 
                       tax_show = 25, 
                       color_vector = c("snow", "steelblue")) + 
                        theme_classic() + 
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))

av2_dlab

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/dlab_symb_hm_samplebyday.pdf", width = 18, height = 13)
av2_dlab
dev.off()
```



##################################################################################
##################################################################################
###                          CORE ASVs w/ AMPVIS2                              ###
##################################################################################
##################################################################################

#amp_venn: Venn diagram of core OTUs

##################################################################################
###                                    PSTR                                    ###
##################################################################################

#PSTR (Samples through time, D1-D2-D3) **W/O SYMBIODINIACEAE**
```{r}
library(ampvis2)

ampvis2_pstr<- amp_load(diel_pstr)

avenn_pstr<- amp_venn(ampvis2_pstr, 
                       group_by = "day_night", 
                       cut_a = 0.1,
                       cut_f = 80, 
                       text_size = 5, 
                       detailed_output = FALSE,
                       normalise = TRUE) + 
                       theme_classic() + 
                       scale_colour_manual(values=c("midnightblue","goldenrod1"))

avenn_pstr

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/heatmaps/pstr_avenn_samplename.pdf", width = 18, height = 13)
avenn_pstr
dev.off()
```




##################################################################################
##################################################################################
###                                 Rhythmicity                                ###
##################################################################################
##################################################################################


#Limma/lmFit requires matrix-like data object containing log ratios or log expression values for a series of arrays
#Preparing the lmFit data
```{r}
library('limorhyde')
library(data.table)

counts_log_pstr <- log(diel_pstr@otu_table+1)
as.data.frame(counts_log_pstr)

meta<- read.delim("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/curacao_mapping.txt", sep ="\t")

meta2<- meta %>% 
  filter(project=="Diel") %>%
  filter(host=="Pseudodiploria strigosa") %>%
  dplyr::select(sample_id, diel_code, diel_day, diel_time) %>%
  mutate(diel_time_num = 
           case_when(
             diel_time=="T12A"~0,
             diel_time=="T6A"~6,
             diel_time=="T12P"~12,
             diel_time=="T6P"~18,
             TRUE~100
           )
           )

meta3<- meta2 %>% 
  mutate(rownames=sample_id) %>%
  column_to_rownames("rownames")

meta3 <- mutate(meta3,
                day = as.numeric(str_remove(diel_day, "D")),
                true_time = diel_time_num + (24*(day-1)))

meta3 = cbind(meta3, limorhyde(meta3$diel_time_num, 'time_'))
```

#LimoRhyde **USING LOG COUNTS**
```{r, fig.wdith = 12}
period = 24
step = 6
qvalRhyCutoff = 0.15
qvalDrCutoff = 0.1


rhyLimma = {
  design = model.matrix(~ time_cos + time_sin, data = meta3)
  fit = lmFit(counts_log_pstr, design)
  fit = eBayes(fit, trend = TRUE)
  rhyNow = data.table(topTable(fit, coef = 2:3, number = Inf), keep.rownames = TRUE)
  setnames(rhyNow, 'rn', 'asv')}
rhyLimma[1:5, ]

write.csv(rhyLimma, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/pstr16s_rhythmictaxa.csv")

rhyLimmaSummary = rhyLimma[, .(P.Value = min(P.Value)), by = asv]
rhyLimmaSummary[, adj.P.Val := p.adjust(P.Value, method = 'BH')]
setorderv(rhyLimmaSummary, 'adj.P.Val')

drLimma = data.table(topTable(fit, coef = 2:3, number = Inf), keep.rownames = TRUE)
setnames(drLimma, 'rn', 'asv')

#Designate the rhythmic taxa based on a p-value of 0.05
rhythmic <- (rhyLimma %>% filter(P.Value <= 0.05))$asv
#Prune the taxa from your phyloseq
diel_pstr_rhythmic <- prune_taxa(rhythmic, diel_pstr)

#Export table for supplemental Section 
#First take the rhylimma products to append to the table
diel_pstr_rhythmic_rhylimma <- data.frame(rhyLimma) 
colnames(diel_pstr_rhythmic_rhylimma)[1] <- "ASV"
diel_pstr_rhythmic_rhylimma <- diel_pstr_rhythmic_rhylimma %>% subset(., select = -adj.P.Val) %>% subset(., ASV %in% rhythmic)
#create the otu and tax tables
diel_pstr_rhythmic_otutable <- data.frame(diel_pstr_rhythmic@otu_table)
diel_pstr_rhythmic_taxtable <- data.frame(diel_pstr_rhythmic@tax_table)
#swap rownames to columns
diel_pstr_rhythmic_otutable <- tibble::rownames_to_column(diel_pstr_rhythmic_otutable, "ASV")
diel_pstr_rhythmic_taxtable <- tibble::rownames_to_column(diel_pstr_rhythmic_taxtable, "ASV")
#create the merged table
rhythmic_rhylimmatax <- merge(diel_pstr_rhythmic_rhylimma, diel_pstr_rhythmic_taxtable, by = "ASV")
rhythmic_otutax <- merge(rhythmic_rhylimmatax, diel_pstr_rhythmic_otutable, by = "ASV")
write.csv(rhythmic_otutax, file = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/pstr16s_rhythmictaxa.csv")

#Set the top 5 most rhythmic
top5rhythmic <- rhythmic[1:5]
#prune taxa for only those top 5
diel_pstr_rhythmic_top5 <- prune_taxa(top5rhythmic, diel_pstr)

#Manipulate the data to add some needed columns for the x axis
meta3 <- mutate(meta3,
                day = as.numeric(str_remove(diel_day, "D")),
                true_time = as.numeric(diel_time_num + (24*(day-1))),
                diel_time_num = as.numeric(diel_time_num))

#Set up the data frame to analyze
asv=t(data.frame(diel_pstr_rhythmic@otu_table))
asv_dat <- cbind(asv,meta3 %>% dplyr::select(diel_time_num, true_time,day))

#or top 5
asv_top5 <- t(data.frame(diel_pstr_rhythmic_top5@otu_table))
asv_top5_dat <- cbind(asv_top5,meta3 %>% dplyr::select(diel_time_num, true_time,day))

#Designate the rects we need to visualize the patterns based on true time
rects <- data.frame(xmin = seq(-3,63,6), xmax = seq(3,69,6), ymin = rep(-Inf, 12), ymax = rep(Inf,12),
              fill = factor(rep(c("12AM","6AM","12PM","6PM"), 3)),levels = c("12AM","6AM","12PM","6PM"))

```


#ASV_547 #Relative Abundance
```{r}
#start plotting the data to visualize the trends
dpr_ra <- diel_pstr_rhythmic
dpr_asv_ra <- transform_sample_counts(dpr_ra, function(x) {x/sum(x)*100})
asv_ra=t(data.frame(diel_pstr_rhythmic@otu_table))
asv_ra_dat <- cbind(asv,meta3 %>% dplyr::select(diel_time_num, true_time,day))


asv_dat_test <- diel_pstr_rhythmic %>% 
  transform_sample_counts(., function(x) {x/sum(x)*100})
asv_dat_test<- t(data.frame(asv_dat_test@otu_table)) %>%
  cbind(.,meta3 %>% dplyr::select(diel_time_num, true_time,day))
  


ASV_547<-ggplot(asv_dat_test, aes(x = true_time, y = ASV_547, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
ASV_547_data <- ggplot_build(ASV_547)$data

ASV_547_mean<-data.frame(ASV_547_data[[2]][["middle"]])

ASV_547_time<-data.frame(ASV_547_data[[2]][["x"]])

ASV_547mean<-cbind(ASV_547_mean,ASV_547_time)

colnames(ASV_547mean) <- c("mean","time")

ASV_547_title <- "ASV_547: Gammaproteobacteria; Enterobacterales; Pseudoalteromonas_sp."

ASV_547<-ggplot(asv_dat_test, aes(x = true_time, y = ASV_547, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = asv547mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Relative Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_547") +
    ggtitle(label = ASV_547_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv547_RA.pdf", width = 7, height = 5)
ASV_547
dev.off()

```


#ASV_547 #Raw Counts
```{r}
#start plotting the data to visualize the trends
ASV_547<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_547, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
asv_547_data <- ggplot_build(asv_547)$data

asv_547_mean<-data.frame(asv_547_data[[2]][["middle"]])

asv_547_time<-data.frame(asv_547_data[[2]][["x"]])

asv_547mean<-cbind(asv_547_mean,asv_547_time)

colnames(asv_547mean) <- c("mean","time")

asv547_title <- "ASV_547: Gammaproteobacteria; Enterobacterales; Pseudoalteromonas_sp."

asv_547<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_547, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = asv547mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_547") +
    ggtitle(label = asv_547_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv547.pdf", width = 7, height = 5)
ASV_547
dev.off()

```

#ASV_2644
```{r}
#start plotting the data to visualize the trends
ASV_2644<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_2644, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
ASV_2644_data <- ggplot_build(ASV_2644)$data

ASV_2644_mean<-data.frame(ASV_2644_data[[2]][["middle"]])

ASV_2644_time<-data.frame(ASV_2644_data[[2]][["x"]])

ASV_2644mean<-cbind(ASV_2644_mean,ASV_2644_time)

colnames(ASV_2644mean) <- c("mean","time")

ASV_2644_title <- "ASV_2644: Gammaproteobacteria; Xanthomonadales; Rhodanobacter_sp."

ASV_2644<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_2644, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = ASV_2644mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_2644") +
    ggtitle(label = ASV_2644_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv2644.pdf", width = 7, height = 5)
ASV_2644
dev.off()

```


#ASV_2863
```{r}
#start plotting the data to visualize the trends
ASV_2863<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_2863, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
ASV_2863_data <- ggplot_build(ASV_2863)$data

ASV_2863_mean<-data.frame(ASV_2863_data[[2]][["middle"]])

ASV_2863_time<-data.frame(ASV_2863_data[[2]][["x"]])

ASV_2863mean<-cbind(ASV_2863_mean,ASV_2863_time)

colnames(ASV_2863mean) <- c("mean","time")

ASV_2863_title <- "ASV_2863: Clostridia; Lachnospirales; Defluviitaleaceae_UCG-011_sp."

ASV_2863<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_2863, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = ASV_2863mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_2863") +
    ggtitle(label = ASV_2863_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv2863.pdf", width = 7, height = 5)
ASV_2863
dev.off()

```


#ASV_9483
```{r}
#start plotting the data to visualize the trends
ASV_9483<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_9483, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
ASV_9483_data <- ggplot_build(ASV_9483)$data

ASV_9483_mean<-data.frame(ASV_9483_data[[2]][["middle"]])

ASV_9483_time<-data.frame(ASV_9483_data[[2]][["x"]])

ASV_9483mean<-cbind(ASV_9483_mean,ASV_9483_time)

colnames(ASV_9483mean) <- c("mean","time")

ASV_9483_title <- "ASV_9483: Clostridia; Clostridia UCG-014; Clostridia UCG-014_XX_sp."

ASV_9483<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_9483, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = ASV_9483mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_9483") +
    ggtitle(label = ASV_9483_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv9483.pdf", width = 7, height = 5)
ASV_9483
dev.off()

```

#ASV_9875
```{r}
#start plotting the data to visualize the trends
ASV_9875<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_9875, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) 

#Find the mean values to plot geom_line
ASV_9875_data <- ggplot_build(ASV_9875)$data

ASV_9875_mean<-data.frame(ASV_9875_data[[2]][["middle"]])

ASV_9875_time<-data.frame(ASV_9875_data[[2]][["x"]])

ASV_9875mean<-cbind(ASV_9875_mean,ASV_9875_time)

colnames(ASV_9875mean) <- c("mean","time")

ASV_9875_title <- "ASV_9875: Verrucomicrobiae; Chthoniobacterales; Chthoniobacter_sp."

ASV_9875<-ggplot(asv_top5_dat, aes(x = true_time, y = ASV_9875, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = ASV_9875mean, aes(x = time, y = mean), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance") +
    theme_classic() +
    theme(legend.position = "bottom") +
    coord_cartesian(xlim = c(-3,69)) +
    ggtitle("ASV_9875") +
    ggtitle(label = ASV_9875_title)

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/rhythmic/rythmic_asv9875.pdf", width = 7, height = 5)
ASV_9875
dev.off()

```



Line only
```{r}


#Set up the data frame to analyze
asv= data.frame(diel_pstr_rhythmic@otu_table)
asv_dat <- cbind(asv,meta3 %>% dplyr::select(diel_time_num, true_time,day))

#or top 5
asv_top5 <- t(data.frame(diel_pstr_rhythmic_top5@otu_table))
asv_top5_dat <- cbind(asv_top5,meta3 %>% dplyr::select(diel_time_num, true_time,day))


dat <- data.frame(asv_dat = asv_dat["ASV_547",])
  dat <- cbind(dat,meta3 %>% dplyr::select(diel_time_num, true_time,day))
  dat_mean = dat %>% dplyr::group_by(true_time) %>%
  dplyr::summarise(asv_dat = median(asv_dat),true_time = median(true_time), diel_time_num = mean(diel_time_num))
  dat_mean_small = dat %>% dplyr::group_by(diel_time_num) %>%
  dplyr::summarise(asv_dat = median(asv_dat),diel_time_num = mean(diel_time_num))



plot_period <- function(x){
  name <- x
  dat <- data.frame(asv_dat = asv_dat[x,])
  dat <- cbind(dat,meta3 %>% dplyr::select(diel_time_num, true_time,day))
  dat_mean = dat %>% dplyr::group_by(true_time) %>%
  dplyr::summarise(asv = median(asv),true_time = median(true_time), diel_time_num = mean(diel_time_num))
  dat_mean_small = dat %>% dplyr::group_by(diel_time_num) %>%
  dplyr::summarise(asv = median(asv),diel_time_num = mean(diel_time_num))
  
  rects <- data.frame(xmin = seq(-3,63,6), xmax = seq(3,69,6), ymin = rep(-Inf, 12), ymax = rep(Inf,12),
              fill = factor(rep(c("12AM","6AM","12PM","6PM"), 3)),levels = c("12AM","6AM","12PM","6PM"))
  
  a<- ggplot(dat, aes(x = true_time, y = asv, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean, aes(x = true_time, y = asv), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Expression (asv)", title = name) +
    theme_classic() +
    theme(legend.position = "bottom")+
    coord_cartesian(xlim = c(-3,69))
  
  b <- ggplot(dat, aes(x = diel_time_num, y = asv, group = diel_time_num)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean_small, aes(x = diel_time_num, y = asv), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance", title = name) +
    theme_classic() +
    theme(legend.position = "none") +
    coord_cartesian(xlim = c(-3,21))
  
  cowplot::plot_grid(plotlist = list(a,b), rel_widths = c(0.7,0.3))
}

plot_period("ASV_547")
  
```





```{r}
##Trial function to plot ASVs at choosing
plot_period <- function(x){
  name <- x
  dat <- data.frame(asv = asv[x,])
  dat <- cbind(dat,meta3 %>% dplyr::select(diel_time_num, true_time,day))
  dat_mean = dat %>% group_by(true_time) %>%
  dplyr::summarise(asv = median(asv),true_time = median(true_time), diel_time_num = mean(diel_time_num))
  dat_mean_small = dat %>% dply::group_by(diel_time_num) %>%
  dplyr::summarise(asv = median(asv),diel_time_num = mean(diel_time_num))
  
rects <- data.frame(xmin = seq(-3,63,6), xmax = seq(3,69,6), ymin = rep(-Inf, 12), ymax = rep(Inf,12),
              fill = factor(rep(c("12AM","6AM","12PM","6PM"), 3)),levels = c("12AM","6AM","12PM","6PM"))
  
  a<- ggplot(dat, aes(x = true_time, y = asv, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean, aes(x = true_time, y = asv), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Expression (asv)", title = name) +
    theme_classic() +
    theme(legend.position = "bottom")+
    coord_cartesian(xlim = c(-3,69))
  
  b <- ggplot(dat, aes(x = diel_time_num, y = asv, group = diel_time_num)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean_small, aes(x = diel_time_num, y = asv), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Abundance (asv)", title = name) +
    theme_classic() +
    theme(legend.position = "none")+
    coord_cartesian(xlim = c(-3,21))
  
  cowplot::plot_grid(plotlist = list(a,b), rel_widths = c(0.7,0.3))
}

rhyLimma["ASV_139",]

plot_period("ASV_139")
ggsave(filename = "~/Desktop/ASV_139.png")



plot_period <- function(x){
  name <- x
  dat <- data.frame(tpm = tpm[x,])
  dat <- cbind(dat,meta3 %>% dplyr::select(diel_time_num, true_time,day))
  dat_mean = dat %>% group_by(true_time) %>%
    dplyr::summarise(tpm = median(tpm),true_time = median(true_time), diel_time_num = mean(diel_time_num))
  dat_mean_small = dat %>% group_by(diel_time_num) %>%
    dplyr::summarise(tpm = median(tpm),diel_time_num = mean(diel_time_num))
  
rects <- data.frame(xmin = seq(-3,63,6), xmax = seq(3,69,6), ymin = rep(-Inf, 12), ymax = rep(Inf,12),
              fill = factor(rep(c("12AM","6AM","12PM","6PM"), 3)),levels = c("12AM","6AM","12PM","6PM"))
  
  a<- ggplot(dat, aes(x = true_time, y = tpm, group = true_time)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean, aes(x = true_time, y = tpm), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Expression (TPM)", title = name) +
    theme_classic() +
    theme(legend.position = "bottom")+
    coord_cartesian(xlim = c(-3,69))
  
  b <- ggplot(dat, aes(x = diel_time_num, y = tpm, group = diel_time_num)) +
    geom_rect(data = rects, mapping = aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf, fill = fill), inherit.aes = FALSE, alpha = 0.5) +
    geom_boxplot() +
    geom_point() +
    geom_line(data = dat_mean_small, aes(x = diel_time_num, y = tpm), inherit.aes = FALSE) +
    scale_x_continuous(breaks = seq(0,72,6)) +
    scale_fill_manual(values = c("12AM"="midnightblue","6AM"="slategray","12PM"="gold","6PM"="orange")) +
    labs(x = "Time (h)", y = "Expression (TPM)", title = name) +
    theme_classic() +
    theme(legend.position = "none")+
    coord_cartesian(xlim = c(-3,21))
  
  cowplot::plot_grid(plotlist = list(a,b), rel_widths = c(0.7,0.3))
}

rhyLimma["TRINITY_DN5918_c0_g1_i2",]

plot_period("TRINITY_DN7440_c0_g2_i3")
ggsave(filename = "~/Desktop/TRINITY_DN7440_c0_g2_i3_period.png")

pdf("/Volumes/projects/weiler/rna_seq/analysis/rhythmic_TRINITY_DN7440_c0_g2_i3_period.pdf", width = 12, height = 7)
plot_period("TRINITY_DN7440_c0_g2_i3")
dev.off()



```

```{r}

##get median expression at each diel_timepoint of rhythmic genes
test <- counts_log_pstr[rhythmic,] %>% as.data.frame() %>%
  rownames_to_column("asv") %>%
  pivot_longer(cols = -asv, names_to = "sample_id", values_to = "clog") %>% #clog is log scale of counts
  inner_join(map %>% dplyr::select(diel_time_num, library_name)) %>%
  group_by(diel_time_num, asv) %>%
  dplyr::summarise(median_clog = median(clog)) %>%
  pivot_wider(id_cols = asv, names_from = diel_time_num, values_from = median_clog) %>%
  column_to_rownames("asv") %>%
  as.matrix()

##calculate which rhythmic genes have a log2 fold change between max and min median expression, ie log2 amplitude > 1
l2fc_pass <- (log(rowMaxs(test),2) - log(rowMins(test),2)) > 1


l2fc_pass[l2fc_pass==TRUE] %>% length()

test[l2fc_pass,]

TRINITY_DN1060_c0_g2_i1



```





```{r, fig.width=10, fig.height = 12}
library(ComplexHeatmap)

meta3 <- meta3 %>% arrange(true_time)
time_order <- (meta3 %>% dplyr::select(sample_id, true_time) %>% arrange(true_time))$sample_id

mat <- counts_log[rhythmic[l2fc_pass],time_order] %>% t() %>% scale() %>% t() #scaled expression of sig genes for Complex heatmap
nrow(mat)

column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              true_time = meta3$true_time,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$true_time
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic.pdf", width = 12, height = 7)
hm
dev.off()


meta3 <- meta3 %>% arrange(true_time)
time_order <- (meta3 %>% dplyr::select(sample_id, true_time) %>% arrange(true_time))$sample_id

mat <- tpm[rhythmic[l2fc_pass],time_order] %>% t() %>% scale() %>% t() #scaled expression of sig genes for Complex heatmap
nrow(mat)

column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              true_time = meta3$true_time,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$true_time
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic.pdf", width = 12, height = 7)
hm
dev.off()





column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$diel_day
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic_day_clog.pdf", width = 12, height = 7)
hm
dev.off()
```


##################################################################################
###                               Symbiodiniaceae                              ###
##################################################################################
#Limma/lmFit requires matrix-like data object containing log ratios or log expression values for a series of arrays
#Preparing the lmFit data
```{r}
library('limorhyde')

counts_log_pstr <- log(dielsymb_pstr@otu_table+1)
as.data.frame(counts_log_pstr)

meta<- read.delim("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/curacao_mapping.txt", sep ="\t")

meta2<- meta %>% 
  filter(project=="Diel") %>%
  filter(host=="Pseudodiploria strigosa") %>%
  dplyr::select(sample_id, diel_code, diel_day, diel_time) %>%
  mutate(diel_time_num = 
           case_when(
             diel_time=="T12A"~0,
             diel_time=="T6A"~6,
             diel_time=="T12P"~12,
             diel_time=="T6P"~18,
             TRUE~100
           )
           )

meta3<- meta2 %>% 
  mutate(rownames=sample_id) %>%
  column_to_rownames("rownames")

meta3 <- mutate(meta3,
                day = as.numeric(str_remove(diel_day, "D")),
                true_time = diel_time_num + (24*(day-1)))

meta3 = cbind(meta3, limorhyde(meta3$diel_time_num, 'time_'))
```

#LimoRhyde **USING LOG COUNTS**
```{r, fig.wdith = 12}
period = 24
step = 6
qvalRhyCutoff = 0.15
qvalDrCutoff = 0.1


rhyLimma = {
  design = model.matrix(~ time_cos + time_sin, data = meta3)
  fit = lmFit(counts_log_pstr, design)
  fit = eBayes(fit, trend = TRUE)
  rhyNow = data.table(topTable(fit, coef = 2:3, number = Inf), keep.rownames = TRUE)
  setnames(rhyNow, 'rn', 'asv')}
rhyLimma[1:5, ]


rhyLimmaSummary = rhyLimma[, .(P.Value = min(P.Value)), by = asv]
rhyLimmaSummary[, adj.P.Val := p.adjust(P.Value, method = 'BH')]
setorderv(rhyLimmaSummary, 'adj.P.Val')

drLimma = data.table(topTable(fit, coef = 2:3, number = Inf), keep.rownames = TRUE)
setnames(drLimma, 'rn', 'asv')

rhythmic <- (rhyLimma %>% filter(adj.P.Val <= 0.01))$asv

##get median expression at each diel_timepoint of rhythmic genes
test <- counts_log[rhythmic,] %>% as.data.frame() %>%
  rownames_to_column("asv") %>%
  pivot_longer(cols = -asv, names_to = "sample_id", values_to = "clog") %>% #clog is log scale of counts
  inner_join(map %>% dplyr::select(diel_time_num, sample_id)) %>%
  group_by(diel_time_num, asv) %>%
  dplyr::summarise(median_clog = median(clog)) %>%
  pivot_wider(id_cols = asv, names_from = diel_time_num, values_from = median_clog) %>%
  column_to_rownames("asv") %>%
  as.matrix()

##calculate which rhythmic genes have a log2 fold change between max and min median expression, ie log2 amplitude > 1
l2fc_pass <- (log(rowMaxs(test),2) - log(rowMins(test),2)) > 1


l2fc_pass[l2fc_pass==TRUE] %>% length()

test[l2fc_pass,]

TRINITY_DN1060_c0_g2_i1


meta3 <- mutate(meta3,
                day = as.numeric(str_remove(diel_day, "D")),
                true_time = diel_time_num + (24*(day-1)))
```





```{r, fig.width=10, fig.height = 12}
library(ComplexHeatmap)

meta3 <- meta3 %>% arrange(true_time)
time_order <- (meta3 %>% dplyr::select(sample_id, true_time) %>% arrange(true_time))$sample_id

mat <- counts_log[rhythmic[l2fc_pass],time_order] %>% t() %>% scale() %>% t() #scaled expression of sig genes for Complex heatmap
nrow(mat)

column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              true_time = meta3$true_time,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$true_time
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic.pdf", width = 12, height = 7)
hm
dev.off()


meta3 <- meta3 %>% arrange(true_time)
time_order <- (meta3 %>% dplyr::select(sample_id, true_time) %>% arrange(true_time))$sample_id

mat <- tpm[rhythmic[l2fc_pass],time_order] %>% t() %>% scale() %>% t() #scaled expression of sig genes for Complex heatmap
nrow(mat)

column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              true_time = meta3$true_time,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$true_time
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic.pdf", width = 12, height = 7)
hm
dev.off()





column_ha <- columnAnnotation(diel_time = meta3$diel_time, 
                              day_night= meta3$day_night,
                              #sample=anno_text(meta3$sample_id), 
                              col=list(
                                diel_time=c("T12A"="midnightblue","T6A"="slategray","T12P"="gold","T6P"="orange"), 
                                day_night=c("Night"="midnightblue","Day"="gold")
                              ))
hm<-Heatmap(mat,
        name = "Scaled Expression",
        top_annotation = column_ha,
        cluster_columns = FALSE,
        #right_annotation = row_ha,
        show_row_names = FALSE,
        column_names_side = "top",
        split = 2,
        column_split = meta3$diel_day
        )

pdf("/Volumes/projects/weiler/rna_seq/analysis/heatmap_daynight_rhythmic_day_clog.pdf", width = 12, height = 7)
hm
dev.off()
```



##################################################################################
##################################################################################
###                               Core Microbiome                              ###
##################################################################################
##################################################################################

```{r}
library(microbiome)
#Need Heuristic to define core microbiome
#prevalence is number of samples it is found in
#detection is percentage of reads per sample
diel_pstr_core <- core_members(diel_pstr, detection = 0.0001, prevalence = 50/100)
diel_pstr_core

core.taxa<- read.csv("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/coretaxa.txt",header=F)

#designate all taxa
diel_pstr_taxa <- taxa_names(diel_pstr)

#remove all but the core taxa
diel_pstr_core_taxa <- diel_pstr_taxa[(diel_pstr_taxa %in% diel_pstr_core)]

#new phyloseq containing only core taxa
diel_pstr_core_members <- prune_taxa(diel_pstr_core_taxa, diel_pstr)

#check the taxonomy
View(diel_pstr_core_members@tax_table)

#save the core taxa to an excel file
write.csv(diel_pstr_core_members@tax_table, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_coremicrobiome_taxtable.csv")
```

#75%
```{r}
library(microbiome)
#Need Heuristic to define core microbiome
#prevalence is number of samples it is found in
#detection is percentage of reads per sample
diel_pstr_core75 <- core_members(diel_pstr, detection = 0.0001, prevalence = 0.75)
diel_pstr_core75

#designate all taxa
diel_pstr_taxa <- taxa_names(diel_pstr)

#remove all but the core taxa
diel_pstr_core_taxa <- diel_pstr_taxa[(diel_pstr_taxa %in% core.taxa)]

#new phyloseq containing only core taxa
diel_pstr_core_members <- prune_taxa(diel_pstr_core_taxa, diel_pstr)

#check the taxonomy
View(diel_pstr_core_members@tax_table)

#save the core taxa to an excel file
write.csv(diel_pstr_core_members@tax_table, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_coremicrobiome_taxtable.csv")
```


##################################################################################
###                       Shared Core w/ Venn Example                          ###
##################################################################################
#From Shetty and Lahti et al. https://microbiome.github.io/tutorials/core_venn.html

```{r}
#install.packages("eulerr") # If not installed
library(eulerr)
library(microbiome)
devtools::install_github('microsud/microbiomeutilities')
library(microbiomeutilities)
```

#We will aim to identify shared core taxa between nationalities.
```{r}
# simple way to count number of samples in each group
table(meta(diel_pstr)$diel_time, useNA = "always")
```

# convert to relative abundances
```{r}
diel_pstr.rel <- microbiome::transform(diel_pstr, "compositional")
```

#Make a list from variable
```{r}
time_points <- unique(as.character(meta(diel_pstr.rel)$diel_time))
print(time_points)
```

#Write a for loop to go through each of the time points one by one and combine identified core taxa into a list.
```{r}
list_core <- c() # an empty object to store information

for (n in time_points){ # for each variable n in DiseaseState
    #print(paste0("Identifying Core Taxa for ", n))
    
    diel_pstr.sub <- subset_samples(diel_pstr.rel, time_points == n) # Choose sample from DiseaseState by n
    
    diel_pstr_core_m <- core_members(diel_pstr.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.001, # 0.001 relative abundance 0.1% rel abund
                           prevalence = 0.75) #in at least 75% of samples
    print(paste0("No. of core taxa in ", n, " : ", length(diel_pstr_core_m))) # print core taxa identified in each DiseaseState.
    list_core[[n]] <- diel_pstr_core_m # add to a list core taxa for each group.
    #print(list_core)
}
```

```{r}
print(list_core)
```


```{r}
# Specify colors and plot venn
# supplying colors in the order they appear in list_core
mycols <- c(T12A="mediumpurple", T6A="#d6e2e9", T12P="#fcf5c7", T6P="lightgoldenrod") 
plot(venn(list_core),
     fills = mycols)
```

#Here, we see that the ids are only names without taxonomic information. This is because the rownames of otu_table and tax_table are usually IDs or sequences.
#The format_to_besthit() function in microbiomeutilites can be useful which add best taxonomic information that is available as demonstrated below.
```{r}
print(list_core)

# use the pseq.rel object created at the begening of this tutorial. 
taxa_names(diel_pstr.rel)[1:5]
```

#DOES NOT WORK BECAUSE 16s!!
```{r}
# format names
diel_pstr.rel.f <- format_to_besthit(diel_pstr.rel)
# check names
taxa_names(diel_pstr.rel.f)[1:5]

```


##################################################################################
###                               Core Venn Diagram                            ###
##################################################################################


```{r}
#install.packages("ggvenn")
library(ggvenn)

#diel_pstr_core_otu <- diel_pstr_core_members@otu_table %>% group_by(diel_pstr_core_members@sam_data$diel_time) %>% as.data.frame()

listcore <- list_core
listcore <- listcore[c("T12A","T6A", "T12P","T6P")]

venn_diel_pstr <- ggvenn(listcore,
                         columns = NULL,
                         show_elements = FALSE,
                         show_percentage = FALSE,
                         digits = 1,
                         fill_color = c("midnightblue","slategray","gold", "orange"),
                         fill_alpha = 0.5,
                         stroke_color = "black",
                         stroke_alpha = 1,
                         stroke_size = 1,
                         stroke_linetype = "solid",
                         set_name_color = "black",
                         set_name_size = 6,
                         text_color = "black",
                         text_size = 4,
                         label_sep = ",",
                         count_column = NULL,
                         show_outside = c("auto"),
                         auto_scale = FALSE
                          )
              

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/coreupset/pstr_venn_core75_lefttoright.pdf", width = 9, height = 7)
venn_diel_pstr
dev.off()
```

#produce unqiue values to excel
```{r}
library(dplyr)
T12A_unique <- list_core$T12A
T6A_unique <- list_core$T6A
T12P_unique <- list_core$T12P
T6P_unique <- list_core$T6P

write.csv(T12A_unique, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_core_asvs_T12A.csv")
write.csv(T6A_unique, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_core_asvs_T6A.csv")
write.csv(T12P_unique, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_core_asvs_T12P.csv")
write.csv(T6P_unique, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16s_diel/16s_core_asvs_T6P.csv")
```



##################################################################################
###                               Core Upset Plot                              ###
##################################################################################

```{r}
#install.packages("UpSetR")
library(UpSetR)
library(MicrobiotaProcess)

#get data to observe upset - using whole data
upsetda <- get_upset(diel_pstr, factorNames="diel_time")

otuda <- t(diel_pstr@otu_table) %>% as.data.frame

sampleda <- map["diel_time"]

head(sampleda)

upsetda2 <- get_upset(obj=otuda, sampleda=sampleda, 
                     factorNames="diel_time")

#Do Not Touch
pstr_upsetr <- UpSetR::upset(upsetda, 
                     sets=c("T6P","T12P","T6A","T12A"), 
                     sets.bar.color = c("orange","gold","slategray","midnightblue"),
                     order.by = "freq", 
                     empty.intersections = "on", 
                     keep.order = TRUE,
                     mainbar.y.label = "ASV Intersection Size"
                     )

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/coreupset/pstr_upset_dielpstr_freq.pdf", width = 9, height = 5)
pstr_upsetr
dev.off()


#make upset from list of core members
library(UpSetR)

pstr_upsetr_core <- UpSetR::upset(fromList(listcore[c("T6P","T12P","T6A","T12A")]), 
                     sets=c("T6P","T12P","T6A","T12A"), 
                     sets.bar.color = c("orange","gold","slategray","midnightblue"),
                     order.by = "freq", 
                     empty.intersections = "on", 
                     keep.order = TRUE,
                     mainbar.y.label = "ASV Intersection Size"
                     )

pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/16s_diel/coreupset/pstr_upset_dielpstr_core75_freq.pdf", width = 9, height = 5)
pstr_upsetr_core
dev.off()
```

#Overcomplicate the aesthetics **incomplete**
```{r}
library(UpSetR)
library(ComplexUpset)

pstr_upsetr <- upset(listcore, fromList(listcore[c("T12A","T6A","T12P","T6P")]), 
                     sets=c("T12A","T6A","T12P","T6P"), 
                     #sets.bar.color = c("orange","midnightblue","gold","slategray"),
                     order.by = "degree", 
                     #empty.intersections = "on", 
                     keep.order = TRUE,
                     mainbar.y.label = "ASV Intersection Size",
                     stripes=c("orange","midnightblue","gold","slategray")
                     )

pstr_upsetr <- upset(a,colnames(a),
                     intersections=list(c("one"),c("one","two"),c("one","two","three","four"),c("one","three","four"),c("one","two","three"),c("three"),c("two"),c("four"),c("three","four"),c("two","four")),
                     sort_intersections=FALSE)
```






#Note: There are limitations to the number of groups that can be plotted with the eulerr::venn(). Check eulerr documentation for more detailed information.





##################################################################################
##################################################################################
###                                Co-Occurence                                ###
##################################################################################
##################################################################################


#Need PS objects to combine for Networks
```{r}
write.csv(diel_pstr@otu_table, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s_otus.csv")
write.csv(diel_pstr@tax_table, "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s_tax.csv")

```


##################################################################################
###                                    Proks                                   ###
##################################################################################

#Chord plots *PROKS*
```{r}
library(microeco)
library(htmlwidgets)
library(circlize)
library(chorddiag)
library(webshot)
meco_diel_pstr <- phyloseq2meco(diel_pstr)
meco_diel_pstr

t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.001, SparCC_simu_num = 100, env_cols= meco_diel_pstr$sample_table$diel_time)

t1$cal_network(network_method="SpiecEasi")

#tnetwork <- t1$cal_network

#got to here and switched to proks/euks

#t1$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_Network/16s_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexf1 <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_Network/16s_network_genus.gexf")
#igraph1 <- gexf.to.igraph(gexf1)


t1$cal_sum_links(taxa_level = "Genus")
t1$plot_sum_links(plot_pos = TRUE, plot_num = 10) # Plot_pos is true, so positive linkages ar eplotted. 

p <- t1$plot_sum_links(plot_pos = TRUE, plot_num = 10)
p$x$option
chord <- chorddiag(p$x$matrix, groupColors=safe_colorblind_palette, groupnameFontsize=8, margin=120, showTicks=FALSE)

#p + canvas.xlim
#chorddiag(p$x$matrix, groupColors=palette, groupnameFontsize=10)

saveWidget(chord, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/16s_diel_pstr_holobiont_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/16s_diel_pstr_holobiont_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/16s_diel_pstr_holobiont_chord.pdf", delay = 0.2)


```



##################################################################################
###                              Proks / Euks                                  ###
##################################################################################

#Chord plots *PROKS+ EUKS W/O SYMBIODINIACEAE*
```{r}
library(microeco)
library(htmlwidgets)
library(circlize)
library(chorddiag)
library(webshot)
meco_diel_pstr <- phyloseq2meco(pstr_nosymb_network)
meco_diel_pstr

t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.001, SparCC_simu_num = 100)
t1$cal_network(network_method="SpiecEasi")

tnetwork <- t1$cal_network

#got to here and switched to proks/euks

#t1$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s18s_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexf1 <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s18s_network_genus.gexf")
#igraph1 <- gexf.to.igraph(gexf1)

#t1$cal_network_attr()
#t1$cal_node_type()
#t1$plot_taxa_roles(use_type = 1)
#t1$cal_eigen()
t1$cal_sum_links(taxa_level = "Genus")
t1$plot_sum_links(plot_pos = TRUE, plot_num = 10) # Plot_pos is true, so positive linkages ar eplotted. 

p <- t1$plot_sum_links(plot_pos = TRUE, plot_num = 10)
p$x$option
chord <- chorddiag(p$x$matrix, groupColors=safe_colorblind_palette, groupnameFontsize=8, margin=120, showTicks=FALSE)

#p + canvas.xlim
#chorddiag(p$x$matrix, groupColors=palette, groupnameFontsize=10)

saveWidget(chord, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_nosymb_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_nosymb_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_nosymb_chord.pdf", delay = 0.2)


```

#Chord plots *PROKS + EUKS W/ SYMBIODINIACEAE*
```{r}
library(microeco)
meco_diel_pstr <- phyloseq2meco(pstr_all_network)
meco_diel_pstr

t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.00001, SparCC_simu_num = 100)
t1$cal_network(network_method="SpiecEasi")

tnetwork <- t1$cal_network

#got to here and switched to proks/euks

t1$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/diel_pstr_holobiont_network_0.00001filt.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexf1 <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s18s_network_genus.gexf")
#igraph1 <- gexf.to.igraph(gexf1)

#t1$cal_network_attr()
#t1$cal_node_type()
#t1$plot_taxa_roles(use_type = 1)
#t1$cal_eigen()
t1$cal_sum_links(taxa_level = "Family")

p <- t1$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = palette) 

p$x$option
chord <- chorddiag(p$x$matrix, groupColors=NULL, groupnameFontsize=8, margin=120, showTicks=FALSE)

#p + canvas.xlim
#chorddiag(p$x$matrix, groupColors=palette, groupnameFontsize=10)

saveWidget(chord, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.pdf", delay = 0.2)

```


#Chord plots *PROKS + EUKS W/ SYMBIODINIACEAE* **PLOTS FAMILY AND GENUS**
```{r fig.height=15,fig.width=15}
library(microeco)
meco_diel_pstr <- phyloseq2meco(pstr_all_network)
meco_diel_pstr

t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.00001, SparCC_simu_num = 100)
t1$cal_network(network_method="SpiecEasi")

t1$cal_sum_links(taxa_level = "Family")


                        
famcolorder<-c("#DDCC77", "mediumorchid", "#C70E7B", "#F4B95A", "lawngreen", "#8FDA04", "#EF7C12",
                        "#D72000", "#007BC3", "#661100", "#54BCD1", "navyblue", "#EFC7E6", "royalblue", "#F4E3C7")
                        
p <- t1$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = famcolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]




#famcols<- c("f__Symbiodiniaceae"="#DDCC77","f__Corallicolidae"="#FC6882",
#              "f__Lachnospiraceae"="#C70E7B","f__Comamonadaceae"="#F4B95A",
#              "f__Rhodobacteraceae"= "lawngreen","f__Pseudomonadaceae"="#8FDA04",
#              "f__Bacteroidaceae"="#EF7C12","f__Ruminococcaceae"="#D72000",
#              "f__Woeseiaceae"="#007BC3","f__Xanthomonadaceae"="#661100",
#              "f__Burkholderiaceae"="#54BCD1","f__Erysipelotrichaceae"="navyblue",
#              "f__Bacillaceae"="#EFC7E6","f__Endozoicomonadaceae"="royalblue",
#              "f__Lactobacillaceae"="#F4E3C7")


pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = famcolorder, grid.col = famcolorder, transparency = 0.35, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()


#By Genus
library(microeco)
t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.00001, SparCC_simu_num = 100)
t1$cal_network(network_method="SpiecEasi")

gencolorder<-c("#DDCC77", "mediumorchid", "#DDCC77", "mediumorchid", "navyblue", "#EF7C12", "#007BC3", "mediumorchid", 
                        "#C70E7B", "#D72000", "mediumorchid", "royalblue", "honeydew2", "limegreen","saddlebrown")

t1$cal_sum_links(taxa_level = "Genus")

p <- t1$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]


gencols<- c("g__Breviolum"="#DDCC77","g__Corallicolidae_COR1"="mediumorchid",
              "g__Durusdinium"="#DDCC77","g__Anthozoaphila"="mediumorchid",
              "g__Pseudomonas"="navyblue","g__Bacteroides"="#EF7C12",
              "g__Woeseia"="#007BC3","g__Corallicolla"="mediumorchid",
              "g__[Ruminococcus] torques group"="#C70E7B",
              "g__Faecalibacterium"="#D72000",
              "g__Corallicolidae_X"="mediumorchid","g__Endozoicomonas"="royalblue",
              "g__Muribaculaceae_X"="honeydew2","g__Ostreobium"="limegreen",
               "g__Asterionella"="saddlebrown")


pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()




# OLD CODE FOR BASIC FIGURE

#chord <- chorddiag(p$x$matrix, groupColors=groupcols, groupnameFontsize=8, margin=120, showTicks=FALSE)

saveWidget(holochord, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/diel_pstr_holobiont_chord_clean_family.pdf", delay = 0.2)

```


#RANDOM OTHER GARBAGE
```{r}
library(microeco)
meco_diel_pstr <- phyloseq2meco(diel_pstr_networksymb)
meco_diel_pstr

t1 <- trans_network$new(dataset = meco_diel_pstr, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0, SparCC_simu_num = 100)
t1$cal_network(network_method="SpiecEasi")

tnetwork <- t1$cal_network

#got to here and switched to proks/euks

t1$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/16s18s_network_genus.gexf")


t1 <- trans_network$new(dataset = meco_diel_pstr, cor_method = "spearman", use_WGCNA_pearson_spearman = TRUE, filter_thres = 0.0001)
t1$cal_network(COR_p_thres = 0.01, COR_optimization = TRUE)
t1$cal_network(COR_p_thres = 0.01, COR_cut = 0.7)
t1$cal_module(method = "cluster_fast_greedy")

t1$cal_network_attr()
t1$res_network_attr

# get node properties
t1$get_node_table(node_roles = TRUE)
```


##################################################################################
###                                 Day/Night                                  ###
##################################################################################

#12AM-6AM **NIGHT**
```{r fig.height=15,fig.width=15}

library(microeco)
meco_pstrnight <- phyloseq2meco(pstrnight)
meco_pstrnight

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstrnight$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstrnight$tax_table),
Kingdom = str_replace(meco_pstrnight$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstrnight$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstrnight$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstrnight$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstrnight$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstrnight$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstrnight$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstrnight$tax_table <- as.matrix(tax.clean)



tnight <- trans_network$new(dataset = meco_pstrnight, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
tnight$cal_network(network_method="SpiecEasi")



tnetwork <- tnight$cal_network

#got to here and switched to proks/euks

#t12a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft12a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
#igrapht12a <- gexf.to.igraph(gexft12a)

#t12a$cal_network_attr()
#t12a$cal_node_type()
#t12a$plot_taxa_roles(use_type = 1)
#t12a$cal_eigen()



tnight$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#DDCC77", 
                        "navyblue", 
                        "royalblue",
                        "#661100",
                        "#EF7C12", 
                        "#DDCC77",
                        "#C70E7B",
                        "#D72000", 
                        "#1BB6AF",
                        "limegreen",
                        "#9FBD25",
                        "goldenrod2",
                        "#D72000")

p <- tnight$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Symbiodiniaceae_X"="#DDCC77", "f__Symbiodiniaceae_X"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Methylobacterium-Methylorubrum"="#1BB6AF", "f__Beijerinckiaceae"="#1BB6AF",
            "g__Ostreobium"="limegreen", "f__Ostreobiaceae"="limegreen",
            "g__Staphylococcus"="#9FBD25", "f__Staphylococcaceae"="#9FBD25",
            "g__Streptococcus"="goldenrod2", "f__Streptococcaceae"="goldenrod2",
            "f__Ruminococcaceae_X"="#D72000", "f__Ruminococcaceae"="#D72000")
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstrnight_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()

```

#Day
```{r fig.height=15,fig.width=15}
library(microeco)
meco_pstrday <- phyloseq2meco(pstrday)
meco_pstrday

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstrday$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstrday$tax_table),
Kingdom = str_replace(meco_pstrday$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstrday$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstrday$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstrday$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstrday$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstrday$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstrday$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstrday$tax_table <- as.matrix(tax.clean)


tday <- trans_network$new(dataset = meco_pstrday, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
tday$cal_network(network_method="SpiecEasi")



tnetwork <- tday$cal_network

#got to here and switched to proks/euks

#t6a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft6a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
#igrapht6a <- gexf.to.igraph(gexft6a)

#t6a$cal_network_attr()
#t6a$cal_node_type()
#t6a$plot_taxa_roles(use_type = 1)
#t6a$cal_eigen()



tday$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#EF7C12",
                        "#DDCC77", 
                        "royalblue",
                        "navyblue", 
                        "#661100",
                        "#D72000", 
                        "#DDCC77",
                        "#C70E7B",
                        "#C70E7B",
                        "mediumturquoise",
                        "royalblue4",
                        "seagreen",
                        "navyblue"
                        )

p <- tday$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Symbiodiniaceae_X"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B",
            "g__Blautia"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Defluviicoccus"= "mediumturquoise", "f__Defluviicoccaceae"="mediumturquoise",
            "g__Hahella"="royalblue4", "f__Hahellaceae"="royalblue4",
            "g__Parabacteroides"="seagreen", "f__Tannerellaceae"="seagreen",
            "g__Pseudomonadales_XX"="navyblue","f__Pseudomonadaceae"="navyblue")
            
  
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstrday_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()

```

##################################################################################
###                          NO SYMBIODINIACEAE                                ###
##################################################################################

#12AM-6AM **NIGHT**
```{r fig.height=15,fig.width=15}

library(microeco)
meco_pstrnightnosymb <- phyloseq2meco(pstrnightnosymb)
meco_pstrnightnosymb

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstrnightnosymb$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstrnightnosymb$tax_table),
Kingdom = str_replace(meco_pstrnightnosymb$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstrnightnosymb$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstrnightnosymb$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstrnightnosymb$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstrnightnosymb$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstrnightnosymb$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstrnightnosymb$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstrnightnosymb$tax_table <- as.matrix(tax.clean)



tnightnosymb <- trans_network$new(dataset = meco_pstrnightnosymb, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
tnightnosymb$cal_network(network_method="SpiecEasi")



#tnetwork <- tnight$cal_network

#got to here and switched to proks/euks

#t12a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft12a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
#igrapht12a <- gexf.to.igraph(gexft12a)

#t12a$cal_network_attr()
#t12a$cal_node_type()
#t12a$plot_taxa_roles(use_type = 1)
#t12a$cal_eigen()



tnightnosymb$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("mediumorchid", 
                        "mediumorchid", 
                        "navyblue", 
                        "mediumorchid", 
                        "#C70E7B",
                        "#EF7C12",
                        "#F4B95A", 
                        "#D72000",
                        "limegreen",
                        "royalblue", 
                        "#C70E7B",
                        "#661100",
                        "#F4D3D3",
                        "grey25",
                        "#54BCD1")

p <- tnightnosymb$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Anthozoaphila"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Corallicolla"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Comamonadaceae_X"="#F4B95A", "f__Comamonadaceae"="#F4B95A",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Ostreobium"="limegreen", "f__Ostreobiaceae"="limegreen",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Lachnospiraceae_X"="#C70E7B", "f__Lachnospiraceae"="#C70E7B",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Acinetobacter"="#F4D3D3", "f_Moraxellaceae"="#F4D3D3",
            "g__Bacteria_XXXXX"="grey25", "f__Bacteria_XXXX"="grey25",
            "g__Burkholderia-Caballeronia-Paraburkholderia"="#54BCD1", "f__Burkholderiaceae"="#54BCD1")
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstrnightnosymb_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()

```

#Day
```{r fig.height=15,fig.width=15}
library(microeco)
meco_pstrdaynosymb <- phyloseq2meco(pstrdaynosymb)
meco_pstrdaynosymb

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstrdaynosymb$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstrdaynosymb$tax_table),
Kingdom = str_replace(meco_pstrdaynosymb$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstrdaynosymb$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstrdaynosymb$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstrdaynosymb$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstrdaynosymb$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstrdaynosymb$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstrdaynosymb$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstrdaynosymb$tax_table <- as.matrix(tax.clean)


tday <- trans_network$new(dataset = meco_pstrdaynosymb, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
tday$cal_network(network_method="SpiecEasi")



tnetwork <- tday$cal_network

#got to here and switched to proks/euks

#t6a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft6a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
#igrapht6a <- gexf.to.igraph(gexft6a)

#t6a$cal_network_attr()
#t6a$cal_node_type()
#t6a$plot_taxa_roles(use_type = 1)
#t6a$cal_eigen()



tday$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#EF7C12",
                        "#DDCC77", 
                        "royalblue",
                        "navyblue", 
                        "#661100",
                        "#D72000", 
                        "#DDCC77",
                        "#C70E7B",
                        "#C70E7B",
                        "mediumturquoise",
                        "royalblue4",
                        "seagreen",
                        "navyblue"
                        )

p <- tday$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Symbiodiniaceae_X"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B",
            "g__Blautia"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Defluviicoccus"= "mediumturquoise", "f__Defluviicoccaceae"="mediumturquoise",
            "g__Hahella"="royalblue4", "f__Hahellaceae"="royalblue4",
            "g__Parabacteroides"="seagreen", "f__Tannerellaceae"="seagreen",
            "g__Pseudomonadales_XX"="navyblue","f__Pseudomonadaceae"="navyblue")
            
  
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstrdaynosymb_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()

```

##################################################################################
###                                By Timepoint                                ###
##################################################################################

#12AM
```{r fig.height=15,fig.width=15}


#filtering step
#type_taxa_net12a <- pstr12a %>%  
#  transform_sample_counts(function(x) {x/sum(x)*100} )  

#t12atest <- type_taxa_net12a@otu_table %>% as.data.frame(., keep.rownames=TRUE)
#t12apurge <- t12atest %>% filter_all(any_vars(is.numeric(.) & . > 0.0001))

#pstr12a_purge <- phyloseq(otu_table(t12apurge, taxa_are_rows = TRUE),
#                          sample_data(pstr12a@sam_data), 
#                          phyloseq::tax_table(pstr12a@tax_table))


library(microeco)
meco_pstr12a <- phyloseq2meco(pstr12a)
meco_pstr12a

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstr12a$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstr12a$tax_table),
Kingdom = str_replace(meco_pstr12a$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstr12a$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstr12a$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstr12a$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstr12a$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstr12a$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstr12a$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstr12a$tax_table <- as.matrix(tax.clean)



#New Method




#Old but works method

t12a <- trans_network$new(dataset = meco_pstr12a, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
t12a$cal_network(network_method="SpiecEasi")



tnetwork <- t12a$cal_network

#got to here and switched to proks/euks

#t12a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft12a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12a_network_genus.gexf")
#igrapht12a <- gexf.to.igraph(gexft12a)

#t12a$cal_network_attr()
#t12a$cal_node_type()
#t12a$plot_taxa_roles(use_type = 1)
#t12a$cal_eigen()



t12a$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#DDCC77", 
                        "royalblue",
                        "navyblue", 
                        "#EF7C12", 
                        "#661100",
                        "#DDCC77",
                        "#C70E7B",
                        "#F4D3D3",
                        "mediumorchid", 
                        "#D72000", 
                        "#9FBD25",
                        "#D72000", 
                        "grey")

p <- t12a$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Symbiodiniaceae"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Acinetobacter"="#F4D3D3", "f_Moraxellaceae"="#F4D3D3",
            "g__Anthozoaphila"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Staphylococcus"="#9FBD25", "f__Staphylococcaceae"="#9FBD25",
            "f__Ruminococcaceae_X"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Alphaproteobacteria_XXX" = "grey"
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12a_holobiont_chord_genus_new.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()



## OLDER WORKING CODE THAT WORKS, DO NOT TOUCH

pt12a <- t12a$plot_sum_links(plot_pos = TRUE, plot_num = 20, color_values = palette) 

pt12a$x$option

chord_t12a <- chorddiag(pt12a$x$matrix, groupColors=palette, groupnameFontsize=8, margin=120, showTicks=FALSE)


saveWidget(chord_t12a, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12a_holobiont_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12a_holobiont_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12a_holobiont_chord.pdf", delay = 0.2)
```

#6AM
```{r fig.height=15,fig.width=15}
library(microeco)
meco_pstr6a <- phyloseq2meco(pstr6a)
meco_pstr6a

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstr6a$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstr6a$tax_table),
Kingdom = str_replace(meco_pstr6a$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstr6a$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstr6a$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstr6a$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstr6a$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstr6a$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstr6a$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstr6a$tax_table <- as.matrix(tax.clean)


t6a <- trans_network$new(dataset = meco_pstr6a, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
t6a$cal_network(network_method="SpiecEasi")



tnetwork <- t6a$cal_network

#got to here and switched to proks/euks

#t6a$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft6a <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6a_network_genus.gexf")
#igrapht6a <- gexf.to.igraph(gexft6a)

#t6a$cal_network_attr()
#t6a$cal_node_type()
#t6a$plot_taxa_roles(use_type = 1)
#t6a$cal_eigen()



t6a$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#DDCC77", 
                        "navyblue", 
                        "royalblue",
                        "limegreen",
                        "#661100",
                        "#EF7C12", 
                        "#F4B95A",
                        "#DDCC77",
                        "#C70E7B",
                        "#D72000", 
                        "#9FBD25",
                        "lightblue",
                        "#D72000"
                        )

p <- t6a$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Ostreobium"="limegreen", "f__Ostreobiaceae"="limegreen",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Comamonadaceae_X"="#F4B95A", "f__Comamonadaceae"="#F4B95A"
            "g__Symbiodiniaceae"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Staphylococcus"="#9FBD25", "f__Staphylococcaceae"="#9FBD25",
            "g__Streptococcus"="darkred", "f__Streptococcaceae"="darkred",
            "f__Ruminococcaceae_X"="#D72000", "f__Ruminococcaceae"="#D72000",
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6a_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()



## OLDER WORKING CODE THAT WORKS, DO NOT TOUCH

pt6a <- t6a$plot_sum_links(plot_pos = TRUE, plot_num = 20, color_values = palette) 

pt6a$x$option

chord_t6a <- chorddiag(pt6a$x$matrix, groupColors=palette, groupnameFontsize=8, margin=120, showTicks=FALSE)


saveWidget(chord_t6a, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6a_holobiont_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6a_holobiont_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6a_holobiont_chord.pdf", delay = 0.2)
```

#12PM
```{r fig.height=15,fig.width=15}
library(microeco)
meco_pstr12p <- phyloseq2meco(pstr12p)
meco_pstr12p

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstr12p$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstr12p$tax_table),
Kingdom = str_replace(meco_pstr12p$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstr12p$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstr12p$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstr12p$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstr12p$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstr12p$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstr12p$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstr12p$tax_table <- as.matrix(tax.clean)


#filtering step

t12p <- trans_network$new(dataset = meco_pstr12p, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 10)



#New Step Using Julia/Flashweave
## Installed many packages, couldnt find julia using R, maybe way in future but for now need to move on.

t12p <- trans_network$new(dataset = meco_pstr12p, cor_method = NULL, taxa_level = "OTU", filter_thres = 0.0001)

t12p$cal_network(network_method = "FlashWeave", FlashWeave_other_para = "alpha=0.01,sensitive=true,heterogeneous=true")


tnetwork <- t12p$cal_network

#got to here and switched to proks/euks

#t12p$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12p_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft12p <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t12p_network_genus.gexf")
#igrapht12p <- gexf.to.igraph(gexft12p)

#t12p$cal_network_attr()
#t12p$cal_node_type()
#t12p$plot_taxa_roles(use_type = 1)
#t12p$cal_eigen()



t12p$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "mediumorchid", 
                        "#DDCC77", 
                        "royalblue",
                        "navyblue", 
                        "#EF7C12", 
                        "#661100",
                        "#DDCC77",
                        "#C70E7B",
                        "#F4D3D3",
                        "mediumorchid", 
                        "#D72000", 
                        "#9FBD25",
                        "#D72000", 
                        "grey")

p <- t12p$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Symbiodiniaceae"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Acinetobacter"="#F4D3D3", "f_Moraxellaceae"="#F4D3D3",
            "g__Anthozoaphila"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Staphylococcus"="#9FBD25", "f__Staphylococcaceae"="#9FBD25",
            "f__Ruminococcaceae_X"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Alphaproteobacteria_XXX" = "grey")
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12p_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()



## OLDER WORKING CODE THAT WORKS, DO NOT TOUCH

pt12p <- t12p$plot_sum_links(plot_pos = TRUE, plot_num = 20, color_values = palette) 

pt12p$x$option

chord_t12p <- chorddiag(pt12p$x$matrix, groupColors=palette, groupnameFontsize=8, margin=120, showTicks=FALSE)


saveWidget(chord_t12p, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12p_holobiont_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12p_holobiont_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr12p_holobiont_chord.pdf", delay = 0.2)
```

#6PM
```{r fig.height=15,fig.width=15}
library(microeco)
meco_pstr6p <- phyloseq2meco(pstr6p)
meco_pstr6p

####################################### Fix Taxonomy Problems   ###########################################

library(stringr)
tax.clean <- meco_pstr6p$tax_table

#write.csv(tax.clean, "~/Desktop/tax.clean.csv")

tax.clean <- data.frame(row.names = row.names(meco_pstr6p$tax_table),
Kingdom = str_replace(meco_pstr6p$tax_table[,1], "D_0__",""),
Phylum = str_replace(meco_pstr6p$tax_table[,2], "D_1__",""),
Class = str_replace(meco_pstr6p$tax_table[,3], "D_2__",""),
Order = str_replace(meco_pstr6p$tax_table[,4], "D_3__",""),
Family = str_replace(meco_pstr6p$tax_table[,5], "D_4__",""),
Genus = str_replace(meco_pstr6p$tax_table[,6], "D_5__",""),
Species = str_replace(meco_pstr6p$tax_table[,7], "D_6__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){
  
  if (tax.clean[i,2] == "p__"){kingdom <- paste(tax.clean[i,1],"NA", sep = "")
  tax.clean[i, 2] <- kingdom
  tax.clean[i, 3] <- paste(kingdom,"_X", sep = "")
  tax.clean[i, 4] <- paste(kingdom,"XX", sep = "")
  tax.clean[i, 5] <- paste(kingdom,"XXX", sep = "")
  tax.clean[i, 6] <- paste(kingdom,"XXXX", sep = "")
  tax.clean[i, 7] <- paste(kingdom,"XXXX_sp.", sep = "")
} else 
  if (tax.clean[i,3] == "c__"){phylum <- paste(tax.clean[i,2], "_X", sep = "")
  tax.clean[i, 3] <- phylum
  tax.clean[i, 4] <- paste(phylum,"X", sep = "")
  tax.clean[i, 5] <- paste(phylum,"XX", sep = "")
  tax.clean[i, 6] <- paste(phylum,"XXX", sep = "")
  tax.clean[i, 7] <- paste(phylum,"XXX_sp.", sep = "")
} else 
  if (tax.clean[i,4] == "o__"){class <- paste(tax.clean[i,3], "_X", sep = "")
  tax.clean[i, 4] <- class
  tax.clean[i, 5] <- paste(class,"X", sep = "")
  tax.clean[i, 6] <- paste(class,"XX", sep = "")
  tax.clean[i, 7] <- paste(class,"XX_sp.", sep = "")
} else 
  if (tax.clean[i,5] == "f__"){order <- paste(tax.clean[i,4], "_X", sep = "")
  tax.clean[i, 5] <- order
  tax.clean[i, 6] <- paste(order,"X", sep = "")
  tax.clean[i, 7] <- paste(order,"X_sp.", sep = "")
} else 
  if (tax.clean[i,6] == "g__"){family <- paste(tax.clean[i,5], "_X", sep = "")
  tax.clean[i, 6] <- family
   tax.clean[i, 7] <- paste(family,"X_sp.", sep = "")
} else 
  if (tax.clean[i,7] == "s__"){tax.clean$Species[i] <- paste(tax.clean$Genus[i], "sp.",sep = "_")}
}
View(tax.clean)


meco_pstr6p$tax_table <- as.matrix(tax.clean)


t6p <- trans_network$new(dataset = meco_pstr6p, cal_cor = "SparCC", taxa_level = "OTU", filter_thres = 0.0001, SparCC_simu_num = 100)
t6p$cal_network(network_method="SpiecEasi")



tnetwork <- t6p$cal_network

#got to here and switched to proks/euks

#t6p$save_network(filepath = "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6p_network_genus.gexf")
# Open in Gephi and save as gexf again, then continue below
#gexft6p <- read.gexf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/data/16S_18S_Network/t6p_network_genus.gexf")
#igrapht6p <- gexf.to.igraph(gexft6p)

#t6p$cal_network_attr()
#t6p$cal_node_type()
#t6p$plot_taxa_roles(use_type = 1)
#t6p$cal_eigen()



t6p$cal_sum_links(taxa_level = "Genus")

gencolorder<-c("#DDCC77", 
                        "#EF7C12", 
                        "mediumorchid", 
                        "#DDCC77",
                        "#D72000", 
                        "royalblue", 
                        "#661100",
                        "saddlebrown",
                        "navyblue",
                        "#DDCC77",
                        "#C70E7B", 
                        "#C70E7B", 
                        "aquamarine3",
                        "seagreen", 
                        "tomato3")

p <- t6p$plot_sum_links(plot_pos = TRUE, plot_num = 15, color_values = gencolorder) 

poption<-p$x$option
chordmat<-p$x$matrix
groupnames <- poption[["groupNames"]]
list(groupnames)

gencols<- c("g__Breviolum"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Bacteroides"="#EF7C12", "f__Bacteroidaceae"="#EF7C12",
            "g__Corallicolidae_COR1"="mediumorchid", "f__Corallicolidae"="mediumorchid",
            "g__Durusdinium"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__Faecalibacterium"="#D72000", "f__Ruminococcaceae"="#D72000",
            "g__Endozoicomonas"="royalblue", "f__Endozoicomonadaceae"="royalblue",
            "g__Stenotrophomonas" ="#661100", "f__Xanthomonadaceae"="#661100",
            "g__Asterionella"="saddlebrown", "f__Tabellariaceae"="saddlebrown",
            "g__Pseudomonas"="navyblue","f__Pseudomonadaceae"="navyblue",
            "g__Symbiodiniaceae_X"="#DDCC77", "f__Symbiodiniaceae"="#DDCC77",
            "g__[Ruminococcus] torques group"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Blautia"="#C70E7B", "f__Lachnospiraceae"="#C70E7B"
            "g__Clostridium sensu stricto 1"="aquamarine3", "f__Clostridiaceae"="aquamarine3"
            "g__Parabacteroides"="seagreen", "f__Tannerellaceae"="seagreen",
            "g__Parasutterella"="tomato3", "f__Sutterellaceae"="tomato3", 
            
            
pdf("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6p_holobiont_chord_genus.pdf", width = 15, height = 15)
chordDiagram(chordmat, column.col = gencolorder, grid.col = gencolorder, transparency = 0.50, annotationTrack = "grid", 
    preAllocateTracks = list(track.height = max(strwidth(unlist(dimnames(chordmat))))))
# gotta customize those labels we removed above
circos.track(track.index = 1, panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
        facing = "clockwise", niceFacing = TRUE, adj = c(0, 0.5))
}, bg.border = NA) 
dev.off()



## OLDER WORKING CODE THAT WORKS, DO NOT TOUCH

pt6p <- t6p$plot_sum_links(plot_pos = TRUE, plot_num = 20, color_values = palette) 

pt6p$x$option

chord_t6p <- chorddiag(pt6p$x$matrix, groupColors=palette, groupnameFontsize=8, margin=120, showTicks=FALSE)


saveWidget(chord_t6p, file="/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6p_holobiont_chord.html")

webshot("/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6p_holobiont_chord.html" , "/Users/bradweiler/Google Drive/My Drive/curacao/metabarcoding/analysis/diel/network analyses/pstr6p_holobiont_chord.pdf", delay = 0.2)
```



```{r}
net_single <- netConstruct(ps_filtered,
                           dataType = "counts",
                           filtTax = "highestFreq",
                           filtTaxPar = list(highestFreq = 100),
                           measure = "spieceasi",
                           normMethod = "clr", 
                           zeroMethod = "none",
                           sparsMethod = "none",
                           verbose = 3,
                           seed = 123456)
```

```{r}
props_single <- netAnalyze(net_single, 
                           centrLCC = TRUE,
                           clustMethod = "cluster_fast_greedy",
                           hubPar = "eigenvector",
                           weightDeg = FALSE, normDeg = FALSE)
```


```{r}
pdf("PATH", width = 20, height = 20)
p <- plot(props_single, 
          layout=layout_in_circle,
          nodeColor = "cluster", 
          nodeSize = "eigenvector",
          title1 = "Network on ASV level with SPIEC-EASI associations", 
          showTitle = TRUE,
          cexTitle = 2.3)

legend(0.7, 1.1, cex = 2.2, title = "estimated association:",
       legend = c("+","-"), lty = 1, lwd = 3, col = c("#009900","red"), 
       bty = "n", horiz = TRUE)
dev.off()
```


##################################################################################
##################################################################################
##                                 Checkpoint                                   ##
##################################################################################
##################################################################################

#Good chance to save the pipeline
```{r}
save.image("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/data/pstr_16s_diel/16s_diel_finalanalyses.RData")

load("/Users/bradleyweiler/workdrive/My Drive/Weiler et al. Diel PSTR (Data)/data/pstr_16s_diel/16s_diel_finalanalyses.RData")
```

























